// asposewordscloud/dist/model/availableFontsResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const fontInfo_1 = require("./fontInfo");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapAvailableFontsResponse = {
    FontInfo: fontInfo_1.FontInfo,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * The list of fonts, available for document processing.
 */
class AvailableFontsResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(AvailableFontsResponse.attributeTypeMap);
    }
}
exports.AvailableFontsResponse = AvailableFontsResponse;
/**
 * Attribute type map
 */
AvailableFontsResponse.attributeTypeMap = [
    {
        name: "additionalFonts",
        baseName: "AdditionalFonts",
        type: "Array<FontInfo>",
    },
    {
        name: "customFonts",
        baseName: "CustomFonts",
        type: "Array<FontInfo>",
    },
    {
        name: "systemFonts",
        baseName: "SystemFonts",
        type: "Array<FontInfo>",
    }
];
