// asposewordscloud/dist/model/classificationResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const classificationResult_1 = require("./classificationResult");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapClassificationResponse = {
    ClassificationResult: classificationResult_1.ClassificationResult,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: PUT https://api.aspose.cloud/v4.0/words/classify.
 */
class ClassificationResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(ClassificationResponse.attributeTypeMap);
    }
}
exports.ClassificationResponse = ClassificationResponse;
/**
 * Attribute type map
 */
ClassificationResponse.attributeTypeMap = [
    {
        name: "bestClassName",
        baseName: "BestClassName",
        type: "string",
    },
    {
        name: "bestClassProbability",
        baseName: "BestClassProbability",
        type: "number",
    },
    {
        name: "bestResults",
        baseName: "BestResults",
        type: "Array<ClassificationResult>",
    }
];
