// asposewordscloud/dist/model/commentsResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const commentsCollection_1 = require("./commentsCollection");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapCommentsResponse = {
    CommentsCollection: commentsCollection_1.CommentsCollection,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/comments.
 */
class CommentsResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(CommentsResponse.attributeTypeMap);
    }
}
exports.CommentsResponse = CommentsResponse;
/**
 * Attribute type map
 */
CommentsResponse.attributeTypeMap = [
    {
        name: "comments",
        baseName: "Comments",
        type: "CommentsCollection",
    }
];
