// asposewordscloud/dist/model/comment.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const commentLink_1 = require("./commentLink");
const documentPosition_1 = require("./documentPosition");
const storyChildNodes_1 = require("./storyChildNodes");
exports.importsMapComment = {
    CommentLink: commentLink_1.CommentLink,
    DocumentPosition: documentPosition_1.DocumentPosition,
    StoryChildNodes: storyChildNodes_1.StoryChildNodes,
};
/**
 * Comment.
 */
class Comment extends commentLink_1.CommentLink {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(Comment.attributeTypeMap);
    }
}
exports.Comment = Comment;
/**
 * Attribute type map
 */
Comment.attributeTypeMap = [
    {
        name: "author",
        baseName: "Author",
        type: "string",
    },
    {
        name: "content",
        baseName: "Content",
        type: "StoryChildNodes",
    },
    {
        name: "dateTime",
        baseName: "DateTime",
        type: "Date",
    },
    {
        name: "initial",
        baseName: "Initial",
        type: "string",
    },
    {
        name: "rangeEnd",
        baseName: "RangeEnd",
        type: "DocumentPosition",
    },
    {
        name: "rangeStart",
        baseName: "RangeStart",
        type: "DocumentPosition",
    },
    {
        name: "text",
        baseName: "Text",
        type: "string",
    }
];
