// asposewordscloud/dist/model/commentBase.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const documentPosition_1 = require("./documentPosition");
exports.importsMapCommentBase = {
    DocumentPosition: documentPosition_1.DocumentPosition,
};
/**
 * Comment.
 */
class CommentBase {
    constructor(init) {
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return CommentBase.attributeTypeMap;
    }
}
exports.CommentBase = CommentBase;
/**
 * Attribute type map
 */
CommentBase.attributeTypeMap = [
    {
        name: "rangeStart",
        baseName: "RangeStart",
        type: "DocumentPosition",
    },
    {
        name: "rangeEnd",
        baseName: "RangeEnd",
        type: "DocumentPosition",
    },
    {
        name: "author",
        baseName: "Author",
        type: "string",
    },
    {
        name: "initial",
        baseName: "Initial",
        type: "string",
    },
    {
        name: "dateTime",
        baseName: "DateTime",
        type: "Date",
    },
    {
        name: "text",
        baseName: "Text",
        type: "string",
    }
];
