// asposewordscloud/dist/model/compareData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const compareOptions_1 = require("./compareOptions");
exports.importsMapCompareData = {
    CompareOptions: compareOptions_1.CompareOptions,
};
/**
 * Container class for compare documents.
 */
class CompareData {
    constructor(init) {
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return CompareData.attributeTypeMap;
    }
}
exports.CompareData = CompareData;
/**
 * Attribute type map
 */
CompareData.attributeTypeMap = [
    {
        name: "comparingWithDocument",
        baseName: "ComparingWithDocument",
        type: "string",
    },
    {
        name: "author",
        baseName: "Author",
        type: "string",
    },
    {
        name: "dateTime",
        baseName: "DateTime",
        type: "Date",
    },
    {
        name: "compareOptions",
        baseName: "CompareOptions",
        type: "CompareOptions",
    }
];
