// asposewordscloud/dist/model/docSaveOptionsData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const saveOptionsData_1 = require("./saveOptionsData");
exports.importsMapDocSaveOptionsData = {
    SaveOptionsData: saveOptionsData_1.SaveOptionsData,
};
/**
 * container class for doc/dot save options.
 */
class DocSaveOptionsData extends saveOptionsData_1.SaveOptionsData {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(DocSaveOptionsData.attributeTypeMap);
    }
}
exports.DocSaveOptionsData = DocSaveOptionsData;
/**
 * Attribute type map
 */
DocSaveOptionsData.attributeTypeMap = [
    {
        name: "alwaysCompressMetafiles",
        baseName: "AlwaysCompressMetafiles",
        type: "boolean",
    },
    {
        name: "password",
        baseName: "Password",
        type: "string",
    },
    {
        name: "savePictureBullet",
        baseName: "SavePictureBullet",
        type: "boolean",
    },
    {
        name: "saveRoutingSlip",
        baseName: "SaveRoutingSlip",
        type: "boolean",
    }
];
// tslint:enable:quotemark
