// asposewordscloud/dist/model/documentProperties.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const documentProperty_1 = require("./documentProperty");
const linkElement_1 = require("./linkElement");
const wordsApiLink_1 = require("./wordsApiLink");
exports.importsMapDocumentProperties = {
    DocumentProperty: documentProperty_1.DocumentProperty,
    LinkElement: linkElement_1.LinkElement,
    WordsApiLink: wordsApiLink_1.WordsApiLink,
};
/**
 * Collection of document properties.
 */
class DocumentProperties extends linkElement_1.LinkElement {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(DocumentProperties.attributeTypeMap);
    }
}
exports.DocumentProperties = DocumentProperties;
/**
 * Attribute type map
 */
DocumentProperties.attributeTypeMap = [
    {
        name: "list",
        baseName: "List",
        type: "Array<DocumentProperty>",
    }
];
