// asposewordscloud/dist/model/documentPropertiesResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const documentProperties_1 = require("./documentProperties");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapDocumentPropertiesResponse = {
    DocumentProperties: documentProperties_1.DocumentProperties,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling:  GET /documentProperties.
 */
class DocumentPropertiesResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(DocumentPropertiesResponse.attributeTypeMap);
    }
}
exports.DocumentPropertiesResponse = DocumentPropertiesResponse;
/**
 * Attribute type map
 */
DocumentPropertiesResponse.attributeTypeMap = [
    {
        name: "documentProperties",
        baseName: "DocumentProperties",
        type: "DocumentProperties",
    }
];
