// asposewordscloud/dist/model/documentStatData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const footnotesStatData_1 = require("./footnotesStatData");
const pageStatData_1 = require("./pageStatData");
exports.importsMapDocumentStatData = {
    FootnotesStatData: footnotesStatData_1.FootnotesStatData,
    PageStatData: pageStatData_1.PageStatData,
};
/**
 * Container for the document's statistical data.
 */
class DocumentStatData {
    constructor(init) {
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return DocumentStatData.attributeTypeMap;
    }
}
exports.DocumentStatData = DocumentStatData;
/**
 * Attribute type map
 */
DocumentStatData.attributeTypeMap = [
    {
        name: "wordCount",
        baseName: "WordCount",
        type: "number",
    },
    {
        name: "paragraphCount",
        baseName: "ParagraphCount",
        type: "number",
    },
    {
        name: "pageCount",
        baseName: "PageCount",
        type: "number",
    },
    {
        name: "footnotesStatData",
        baseName: "FootnotesStatData",
        type: "FootnotesStatData",
    },
    {
        name: "pageStatData",
        baseName: "PageStatData",
        type: "Array<PageStatData>",
    }
];
