// asposewordscloud/dist/model/documentPropertyResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const documentProperty_1 = require("./documentProperty");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapDocumentPropertyResponse = {
    DocumentProperty: documentProperty_1.DocumentProperty,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling:  GET documentProperties/{propertyName}.
 */
class DocumentPropertyResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(DocumentPropertyResponse.attributeTypeMap);
    }
}
exports.DocumentPropertyResponse = DocumentPropertyResponse;
/**
 * Attribute type map
 */
DocumentPropertyResponse.attributeTypeMap = [
    {
        name: "documentProperty",
        baseName: "DocumentProperty",
        type: "DocumentProperty",
    }
];
