// asposewordscloud/dist/model/documentProperty.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const linkElement_1 = require("./linkElement");
const wordsApiLink_1 = require("./wordsApiLink");
exports.importsMapDocumentProperty = {
    LinkElement: linkElement_1.LinkElement,
    WordsApiLink: wordsApiLink_1.WordsApiLink,
};
/**
 * Words document property DTO.
 */
class DocumentProperty extends linkElement_1.LinkElement {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(DocumentProperty.attributeTypeMap);
    }
}
exports.DocumentProperty = DocumentProperty;
/**
 * Attribute type map
 */
DocumentProperty.attributeTypeMap = [
    {
        name: "builtIn",
        baseName: "BuiltIn",
        type: "boolean",
    },
    {
        name: "name",
        baseName: "Name",
        type: "string",
    },
    {
        name: "value",
        baseName: "Value",
        type: "string",
    }
];
