// asposewordscloud/dist/model/drawingObjectUpdate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.importsMapDrawingObjectUpdate = {};
/**
 * Drawing object element for update.
 */
class DrawingObjectUpdate {
    constructor(init) {
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return DrawingObjectUpdate.attributeTypeMap;
    }
}
exports.DrawingObjectUpdate = DrawingObjectUpdate;
/**
 * Attribute type map
 */
DrawingObjectUpdate.attributeTypeMap = [
    {
        name: "relativeHorizontalPosition",
        baseName: "RelativeHorizontalPosition",
        type: "DrawingObjectUpdate.RelativeHorizontalPositionEnum",
    },
    {
        name: "left",
        baseName: "Left",
        type: "number",
    },
    {
        name: "relativeVerticalPosition",
        baseName: "RelativeVerticalPosition",
        type: "DrawingObjectUpdate.RelativeVerticalPositionEnum",
    },
    {
        name: "top",
        baseName: "Top",
        type: "number",
    },
    {
        name: "width",
        baseName: "Width",
        type: "number",
    },
    {
        name: "height",
        baseName: "Height",
        type: "number",
    },
    {
        name: "wrapType",
        baseName: "WrapType",
        type: "DrawingObjectUpdate.WrapTypeEnum",
    }
];
/**
 * Enums for DrawingObjectUpdate
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (DrawingObjectUpdate) {
    let RelativeHorizontalPositionEnum;
    (function (RelativeHorizontalPositionEnum) {
        RelativeHorizontalPositionEnum[RelativeHorizontalPositionEnum["Margin"] = 'Margin'] = "Margin";
        RelativeHorizontalPositionEnum[RelativeHorizontalPositionEnum["Page"] = 'Page'] = "Page";
        RelativeHorizontalPositionEnum[RelativeHorizontalPositionEnum["Column"] = 'Column'] = "Column";
        RelativeHorizontalPositionEnum[RelativeHorizontalPositionEnum["Default"] = 'Default'] = "Default";
        RelativeHorizontalPositionEnum[RelativeHorizontalPositionEnum["Character"] = 'Character'] = "Character";
        RelativeHorizontalPositionEnum[RelativeHorizontalPositionEnum["LeftMargin"] = 'LeftMargin'] = "LeftMargin";
        RelativeHorizontalPositionEnum[RelativeHorizontalPositionEnum["RightMargin"] = 'RightMargin'] = "RightMargin";
        RelativeHorizontalPositionEnum[RelativeHorizontalPositionEnum["InsideMargin"] = 'InsideMargin'] = "InsideMargin";
        RelativeHorizontalPositionEnum[RelativeHorizontalPositionEnum["OutsideMargin"] = 'OutsideMargin'] = "OutsideMargin";
    })(RelativeHorizontalPositionEnum = DrawingObjectUpdate.RelativeHorizontalPositionEnum || (DrawingObjectUpdate.RelativeHorizontalPositionEnum = {}));
    let RelativeVerticalPositionEnum;
    (function (RelativeVerticalPositionEnum) {
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["Margin"] = 'Margin'] = "Margin";
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["TableDefault"] = 'TableDefault'] = "TableDefault";
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["Page"] = 'Page'] = "Page";
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["Paragraph"] = 'Paragraph'] = "Paragraph";
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["TextFrameDefault"] = 'TextFrameDefault'] = "TextFrameDefault";
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["Line"] = 'Line'] = "Line";
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["TopMargin"] = 'TopMargin'] = "TopMargin";
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["BottomMargin"] = 'BottomMargin'] = "BottomMargin";
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["InsideMargin"] = 'InsideMargin'] = "InsideMargin";
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["OutsideMargin"] = 'OutsideMargin'] = "OutsideMargin";
    })(RelativeVerticalPositionEnum = DrawingObjectUpdate.RelativeVerticalPositionEnum || (DrawingObjectUpdate.RelativeVerticalPositionEnum = {}));
    let WrapTypeEnum;
    (function (WrapTypeEnum) {
        WrapTypeEnum[WrapTypeEnum["Inline"] = 'Inline'] = "Inline";
        WrapTypeEnum[WrapTypeEnum["TopBottom"] = 'TopBottom'] = "TopBottom";
        WrapTypeEnum[WrapTypeEnum["Square"] = 'Square'] = "Square";
        WrapTypeEnum[WrapTypeEnum["None"] = 'None'] = "None";
        WrapTypeEnum[WrapTypeEnum["Tight"] = 'Tight'] = "Tight";
        WrapTypeEnum[WrapTypeEnum["Through"] = 'Through'] = "Through";
    })(WrapTypeEnum = DrawingObjectUpdate.WrapTypeEnum || (DrawingObjectUpdate.WrapTypeEnum = {}));
})(DrawingObjectUpdate = exports.DrawingObjectUpdate || (exports.DrawingObjectUpdate = {}));
// tslint:enable:quotemark
