// asposewordscloud/dist/model/drawingObjectsResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const drawingObjectCollection_1 = require("./drawingObjectCollection");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapDrawingObjectsResponse = {
    DrawingObjectCollection: drawingObjectCollection_1.DrawingObjectCollection,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling:  GET /drawingObjects.
 */
class DrawingObjectsResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(DrawingObjectsResponse.attributeTypeMap);
    }
}
exports.DrawingObjectsResponse = DrawingObjectsResponse;
/**
 * Attribute type map
 */
DrawingObjectsResponse.attributeTypeMap = [
    {
        name: "drawingObjects",
        baseName: "DrawingObjects",
        type: "DrawingObjectCollection",
    }
];
