// asposewordscloud/dist/model/drawingObject.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const drawingObjectLink_1 = require("./drawingObjectLink");
const wordsApiLink_1 = require("./wordsApiLink");
exports.importsMapDrawingObject = {
    DrawingObjectLink: drawingObjectLink_1.DrawingObjectLink,
    WordsApiLink: wordsApiLink_1.WordsApiLink,
};
/**
 * Represents Drawing Object DTO.
 */
class DrawingObject extends drawingObjectLink_1.DrawingObjectLink {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(DrawingObject.attributeTypeMap);
    }
}
exports.DrawingObject = DrawingObject;
/**
 * Attribute type map
 */
DrawingObject.attributeTypeMap = [
    {
        name: "height",
        baseName: "Height",
        type: "number",
    },
    {
        name: "imageDataLink",
        baseName: "ImageDataLink",
        type: "WordsApiLink",
    },
    {
        name: "left",
        baseName: "Left",
        type: "number",
    },
    {
        name: "oleDataLink",
        baseName: "OleDataLink",
        type: "WordsApiLink",
    },
    {
        name: "relativeHorizontalPosition",
        baseName: "RelativeHorizontalPosition",
        type: "DrawingObject.RelativeHorizontalPositionEnum",
    },
    {
        name: "relativeVerticalPosition",
        baseName: "RelativeVerticalPosition",
        type: "DrawingObject.RelativeVerticalPositionEnum",
    },
    {
        name: "renderLinks",
        baseName: "RenderLinks",
        type: "Array<WordsApiLink>",
    },
    {
        name: "top",
        baseName: "Top",
        type: "number",
    },
    {
        name: "width",
        baseName: "Width",
        type: "number",
    },
    {
        name: "wrapType",
        baseName: "WrapType",
        type: "DrawingObject.WrapTypeEnum",
    }
];
/**
 * Enums for DrawingObject
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (DrawingObject) {
    let RelativeHorizontalPositionEnum;
    (function (RelativeHorizontalPositionEnum) {
        RelativeHorizontalPositionEnum[RelativeHorizontalPositionEnum["Margin"] = 'Margin'] = "Margin";
        RelativeHorizontalPositionEnum[RelativeHorizontalPositionEnum["Page"] = 'Page'] = "Page";
        RelativeHorizontalPositionEnum[RelativeHorizontalPositionEnum["Column"] = 'Column'] = "Column";
        RelativeHorizontalPositionEnum[RelativeHorizontalPositionEnum["Default"] = 'Default'] = "Default";
        RelativeHorizontalPositionEnum[RelativeHorizontalPositionEnum["Character"] = 'Character'] = "Character";
        RelativeHorizontalPositionEnum[RelativeHorizontalPositionEnum["LeftMargin"] = 'LeftMargin'] = "LeftMargin";
        RelativeHorizontalPositionEnum[RelativeHorizontalPositionEnum["RightMargin"] = 'RightMargin'] = "RightMargin";
        RelativeHorizontalPositionEnum[RelativeHorizontalPositionEnum["InsideMargin"] = 'InsideMargin'] = "InsideMargin";
        RelativeHorizontalPositionEnum[RelativeHorizontalPositionEnum["OutsideMargin"] = 'OutsideMargin'] = "OutsideMargin";
    })(RelativeHorizontalPositionEnum = DrawingObject.RelativeHorizontalPositionEnum || (DrawingObject.RelativeHorizontalPositionEnum = {}));
    let RelativeVerticalPositionEnum;
    (function (RelativeVerticalPositionEnum) {
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["Margin"] = 'Margin'] = "Margin";
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["TableDefault"] = 'TableDefault'] = "TableDefault";
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["Page"] = 'Page'] = "Page";
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["Paragraph"] = 'Paragraph'] = "Paragraph";
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["TextFrameDefault"] = 'TextFrameDefault'] = "TextFrameDefault";
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["Line"] = 'Line'] = "Line";
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["TopMargin"] = 'TopMargin'] = "TopMargin";
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["BottomMargin"] = 'BottomMargin'] = "BottomMargin";
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["InsideMargin"] = 'InsideMargin'] = "InsideMargin";
        RelativeVerticalPositionEnum[RelativeVerticalPositionEnum["OutsideMargin"] = 'OutsideMargin'] = "OutsideMargin";
    })(RelativeVerticalPositionEnum = DrawingObject.RelativeVerticalPositionEnum || (DrawingObject.RelativeVerticalPositionEnum = {}));
    let WrapTypeEnum;
    (function (WrapTypeEnum) {
        WrapTypeEnum[WrapTypeEnum["Inline"] = 'Inline'] = "Inline";
        WrapTypeEnum[WrapTypeEnum["TopBottom"] = 'TopBottom'] = "TopBottom";
        WrapTypeEnum[WrapTypeEnum["Square"] = 'Square'] = "Square";
        WrapTypeEnum[WrapTypeEnum["None"] = 'None'] = "None";
        WrapTypeEnum[WrapTypeEnum["Tight"] = 'Tight'] = "Tight";
        WrapTypeEnum[WrapTypeEnum["Through"] = 'Through'] = "Through";
    })(WrapTypeEnum = DrawingObject.WrapTypeEnum || (DrawingObject.WrapTypeEnum = {}));
})(DrawingObject = exports.DrawingObject || (exports.DrawingObject = {}));
// tslint:enable:quotemark
