// asposewordscloud/dist/model/fieldsResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const fieldCollection_1 = require("./fieldCollection");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapFieldsResponse = {
    FieldCollection: fieldCollection_1.FieldCollection,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/{nodePath}/fields.
 */
class FieldsResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(FieldsResponse.attributeTypeMap);
    }
}
exports.FieldsResponse = FieldsResponse;
/**
 * Attribute type map
 */
FieldsResponse.attributeTypeMap = [
    {
        name: "fields",
        baseName: "Fields",
        type: "FieldCollection",
    }
];
