// asposewordscloud/dist/model/footnoteBase.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const documentPosition_1 = require("./documentPosition");
exports.importsMapFootnoteBase = {
    DocumentPosition: documentPosition_1.DocumentPosition,
};
/**
 * Footnote base class.
 */
class FootnoteBase {
    constructor(init) {
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return FootnoteBase.attributeTypeMap;
    }
}
exports.FootnoteBase = FootnoteBase;
/**
 * Attribute type map
 */
FootnoteBase.attributeTypeMap = [
    {
        name: "position",
        baseName: "Position",
        type: "DocumentPosition",
    },
    {
        name: "footnoteType",
        baseName: "FootnoteType",
        type: "FootnoteBase.FootnoteTypeEnum",
    },
    {
        name: "referenceMark",
        baseName: "ReferenceMark",
        type: "string",
    },
    {
        name: "text",
        baseName: "Text",
        type: "string",
    }
];
/**
 * Enums for FootnoteBase
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (FootnoteBase) {
    let FootnoteTypeEnum;
    (function (FootnoteTypeEnum) {
        FootnoteTypeEnum[FootnoteTypeEnum["Footnote"] = 'Footnote'] = "Footnote";
        FootnoteTypeEnum[FootnoteTypeEnum["Endnote"] = 'Endnote'] = "Endnote";
    })(FootnoteTypeEnum = FootnoteBase.FootnoteTypeEnum || (FootnoteBase.FootnoteTypeEnum = {}));
})(FootnoteBase = exports.FootnoteBase || (exports.FootnoteBase = {}));
// tslint:enable:quotemark
