// asposewordscloud/dist/model/footnotesResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const footnoteCollection_1 = require("./footnoteCollection");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapFootnotesResponse = {
    FootnoteCollection: footnoteCollection_1.FootnoteCollection,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/footnotes.
 */
class FootnotesResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(FootnotesResponse.attributeTypeMap);
    }
}
exports.FootnotesResponse = FootnotesResponse;
/**
 * Attribute type map
 */
FootnotesResponse.attributeTypeMap = [
    {
        name: "footnotes",
        baseName: "Footnotes",
        type: "FootnoteCollection",
    }
];
