// asposewordscloud/dist/model/formFieldsResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const formFieldCollection_1 = require("./formFieldCollection");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapFormFieldsResponse = {
    FormFieldCollection: formFieldCollection_1.FormFieldCollection,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/{nodePath}/formfields.
 */
class FormFieldsResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(FormFieldsResponse.attributeTypeMap);
    }
}
exports.FormFieldsResponse = FormFieldsResponse;
/**
 * Attribute type map
 */
FormFieldsResponse.attributeTypeMap = [
    {
        name: "formFields",
        baseName: "FormFields",
        type: "FormFieldCollection",
    }
];
