// asposewordscloud/dist/model/formFieldResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const formField_1 = require("./formField");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapFormFieldResponse = {
    FormField: formField_1.FormField,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/paragraphs/{0}/formfields/{1}.
 */
class FormFieldResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(FormFieldResponse.attributeTypeMap);
    }
}
exports.FormFieldResponse = FormFieldResponse;
/**
 * Attribute type map
 */
FormFieldResponse.attributeTypeMap = [
    {
        name: "formField",
        baseName: "FormField",
        type: "FormField",
    }
];
