// asposewordscloud/dist/model/formFieldTextInput.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const formField_1 = require("./formField");
exports.importsMapFormFieldTextInput = {
    FormField: formField_1.FormField,
};
/**
 * FormField text input element.
 */
class FormFieldTextInput extends formField_1.FormField {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(FormFieldTextInput.attributeTypeMap);
    }
}
exports.FormFieldTextInput = FormFieldTextInput;
/**
 * Attribute type map
 */
FormFieldTextInput.attributeTypeMap = [
    {
        name: "maxLength",
        baseName: "MaxLength",
        type: "number",
    },
    {
        name: "textInputDefault",
        baseName: "TextInputDefault",
        type: "string",
    },
    {
        name: "textInputFormat",
        baseName: "TextInputFormat",
        type: "string",
    },
    {
        name: "textInputType",
        baseName: "TextInputType",
        type: "FormFieldTextInput.TextInputTypeEnum",
    }
];
/**
 * Enums for FormFieldTextInput
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (FormFieldTextInput) {
    let TextInputTypeEnum;
    (function (TextInputTypeEnum) {
        TextInputTypeEnum[TextInputTypeEnum["Regular"] = 'Regular'] = "Regular";
        TextInputTypeEnum[TextInputTypeEnum["Number"] = 'Number'] = "Number";
        TextInputTypeEnum[TextInputTypeEnum["Date"] = 'Date'] = "Date";
        TextInputTypeEnum[TextInputTypeEnum["CurrentDate"] = 'CurrentDate'] = "CurrentDate";
        TextInputTypeEnum[TextInputTypeEnum["CurrentTime"] = 'CurrentTime'] = "CurrentTime";
        TextInputTypeEnum[TextInputTypeEnum["Calculated"] = 'Calculated'] = "Calculated";
    })(TextInputTypeEnum = FormFieldTextInput.TextInputTypeEnum || (FormFieldTextInput.TextInputTypeEnum = {}));
})(FormFieldTextInput = exports.FormFieldTextInput || (exports.FormFieldTextInput = {}));
// tslint:enable:quotemark
