// asposewordscloud/dist/model/formField.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const nodeLink_1 = require("./nodeLink");
exports.importsMapFormField = {
    NodeLink: nodeLink_1.NodeLink,
};
/**
 * FromField.
 */
class FormField extends nodeLink_1.NodeLink {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(FormField.attributeTypeMap);
    }
}
exports.FormField = FormField;
/**
 * Attribute type map
 */
FormField.attributeTypeMap = [
    {
        name: "calculateOnExit",
        baseName: "CalculateOnExit",
        type: "boolean",
    },
    {
        name: "enabled",
        baseName: "Enabled",
        type: "boolean",
    },
    {
        name: "entryMacro",
        baseName: "EntryMacro",
        type: "string",
    },
    {
        name: "exitMacro",
        baseName: "ExitMacro",
        type: "string",
    },
    {
        name: "helpText",
        baseName: "HelpText",
        type: "string",
    },
    {
        name: "name",
        baseName: "Name",
        type: "string",
    },
    {
        name: "ownHelp",
        baseName: "OwnHelp",
        type: "boolean",
    },
    {
        name: "ownStatus",
        baseName: "OwnStatus",
        type: "boolean",
    },
    {
        name: "statusText",
        baseName: "StatusText",
        type: "string",
    }
];
