// asposewordscloud/dist/model/headerFooter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const headerFooterLink_1 = require("./headerFooterLink");
const linkElement_1 = require("./linkElement");
const nodeLink_1 = require("./nodeLink");
exports.importsMapHeaderFooter = {
    HeaderFooterLink: headerFooterLink_1.HeaderFooterLink,
    LinkElement: linkElement_1.LinkElement,
    NodeLink: nodeLink_1.NodeLink,
};
/**
 * Section element.
 */
class HeaderFooter extends headerFooterLink_1.HeaderFooterLink {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(HeaderFooter.attributeTypeMap);
    }
}
exports.HeaderFooter = HeaderFooter;
/**
 * Attribute type map
 */
HeaderFooter.attributeTypeMap = [
    {
        name: "childNodes",
        baseName: "ChildNodes",
        type: "Array<NodeLink>",
    },
    {
        name: "drawingObjects",
        baseName: "DrawingObjects",
        type: "LinkElement",
    },
    {
        name: "paragraphs",
        baseName: "Paragraphs",
        type: "LinkElement",
    }
];
// tslint:enable:quotemark
