// asposewordscloud/dist/model/headerFooterLink.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const linkElement_1 = require("./linkElement");
const wordsApiLink_1 = require("./wordsApiLink");
exports.importsMapHeaderFooterLink = {
    LinkElement: linkElement_1.LinkElement,
    WordsApiLink: wordsApiLink_1.WordsApiLink,
};
/**
 * HeaderFooter link element.
 */
class HeaderFooterLink extends linkElement_1.LinkElement {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(HeaderFooterLink.attributeTypeMap);
    }
}
exports.HeaderFooterLink = HeaderFooterLink;
/**
 * Attribute type map
 */
HeaderFooterLink.attributeTypeMap = [
    {
        name: "type",
        baseName: "Type",
        type: "HeaderFooterLink.TypeEnum",
    }
];
/**
 * Enums for HeaderFooterLink
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (HeaderFooterLink) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["HeaderEven"] = 'HeaderEven'] = "HeaderEven";
        TypeEnum[TypeEnum["HeaderPrimary"] = 'HeaderPrimary'] = "HeaderPrimary";
        TypeEnum[TypeEnum["FooterEven"] = 'FooterEven'] = "FooterEven";
        TypeEnum[TypeEnum["FooterPrimary"] = 'FooterPrimary'] = "FooterPrimary";
        TypeEnum[TypeEnum["HeaderFirst"] = 'HeaderFirst'] = "HeaderFirst";
        TypeEnum[TypeEnum["FooterFirst"] = 'FooterFirst'] = "FooterFirst";
    })(TypeEnum = HeaderFooterLink.TypeEnum || (HeaderFooterLink.TypeEnum = {}));
})(HeaderFooterLink = exports.HeaderFooterLink || (exports.HeaderFooterLink = {}));
// tslint:enable:quotemark
