// asposewordscloud/dist/model/headerFooterLinkCollection.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const headerFooterLink_1 = require("./headerFooterLink");
const linkElement_1 = require("./linkElement");
const wordsApiLink_1 = require("./wordsApiLink");
exports.importsMapHeaderFooterLinkCollection = {
    HeaderFooterLink: headerFooterLink_1.HeaderFooterLink,
    LinkElement: linkElement_1.LinkElement,
    WordsApiLink: wordsApiLink_1.WordsApiLink,
};
/**
 * Collection of links to header/footers.
 */
class HeaderFooterLinkCollection extends linkElement_1.LinkElement {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(HeaderFooterLinkCollection.attributeTypeMap);
    }
}
exports.HeaderFooterLinkCollection = HeaderFooterLinkCollection;
/**
 * Attribute type map
 */
HeaderFooterLinkCollection.attributeTypeMap = [
    {
        name: "list",
        baseName: "List",
        type: "Array<HeaderFooterLink>",
    }
];
