// asposewordscloud/dist/model/headerFootersResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const headerFooterLinkCollection_1 = require("./headerFooterLinkCollection");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapHeaderFootersResponse = {
    HeaderFooterLinkCollection: headerFooterLinkCollection_1.HeaderFooterLinkCollection,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/headersfooters.
 */
class HeaderFootersResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(HeaderFootersResponse.attributeTypeMap);
    }
}
exports.HeaderFootersResponse = HeaderFootersResponse;
/**
 * Attribute type map
 */
HeaderFootersResponse.attributeTypeMap = [
    {
        name: "headerFooters",
        baseName: "HeaderFooters",
        type: "HeaderFooterLinkCollection",
    }
];
