// asposewordscloud/dist/model/headerFooterResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const headerFooter_1 = require("./headerFooter");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapHeaderFooterResponse = {
    HeaderFooter: headerFooter_1.HeaderFooter,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/headersfooters/{0}.
 */
class HeaderFooterResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(HeaderFooterResponse.attributeTypeMap);
    }
}
exports.HeaderFooterResponse = HeaderFooterResponse;
/**
 * Attribute type map
 */
HeaderFooterResponse.attributeTypeMap = [
    {
        name: "headerFooter",
        baseName: "HeaderFooter",
        type: "HeaderFooter",
    }
];
