// asposewordscloud/dist/model/listFormat.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const linkElement_1 = require("./linkElement");
const wordsApiLink_1 = require("./wordsApiLink");
exports.importsMapListFormat = {
    LinkElement: linkElement_1.LinkElement,
    WordsApiLink: wordsApiLink_1.WordsApiLink,
};
/**
 * Paragraph list format element.
 */
class ListFormat extends linkElement_1.LinkElement {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(ListFormat.attributeTypeMap);
    }
}
exports.ListFormat = ListFormat;
/**
 * Attribute type map
 */
ListFormat.attributeTypeMap = [
    {
        name: "isListItem",
        baseName: "IsListItem",
        type: "boolean",
    },
    {
        name: "listId",
        baseName: "ListId",
        type: "number",
    },
    {
        name: "listLevelNumber",
        baseName: "ListLevelNumber",
        type: "number",
    }
];
