// asposewordscloud/dist/model/listInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const linkElement_1 = require("./linkElement");
const listLevels_1 = require("./listLevels");
const style_1 = require("./style");
const wordsApiLink_1 = require("./wordsApiLink");
exports.importsMapListInfo = {
    LinkElement: linkElement_1.LinkElement,
    ListLevels: listLevels_1.ListLevels,
    Style: style_1.Style,
    WordsApiLink: wordsApiLink_1.WordsApiLink,
};
/**
 * Represents a single document list.
 */
class ListInfo extends linkElement_1.LinkElement {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(ListInfo.attributeTypeMap);
    }
}
exports.ListInfo = ListInfo;
/**
 * Attribute type map
 */
ListInfo.attributeTypeMap = [
    {
        name: "isListStyleDefinition",
        baseName: "IsListStyleDefinition",
        type: "boolean",
    },
    {
        name: "isListStyleReference",
        baseName: "IsListStyleReference",
        type: "boolean",
    },
    {
        name: "isMultiLevel",
        baseName: "IsMultiLevel",
        type: "boolean",
    },
    {
        name: "isRestartAtEachSection",
        baseName: "IsRestartAtEachSection",
        type: "boolean",
    },
    {
        name: "listId",
        baseName: "ListId",
        type: "number",
    },
    {
        name: "listLevels",
        baseName: "ListLevels",
        type: "ListLevels",
    },
    {
        name: "style",
        baseName: "Style",
        type: "Style",
    }
];
