// asposewordscloud/dist/model/listInsert.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.importsMapListInsert = {};
/**
 * Insert document to document list.
 */
class ListInsert {
    constructor(init) {
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return ListInsert.attributeTypeMap;
    }
}
exports.ListInsert = ListInsert;
/**
 * Attribute type map
 */
ListInsert.attributeTypeMap = [
    {
        name: "template",
        baseName: "Template",
        type: "ListInsert.TemplateEnum",
    }
];
/**
 * Enums for ListInsert
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (ListInsert) {
    let TemplateEnum;
    (function (TemplateEnum) {
        TemplateEnum[TemplateEnum["BulletDefault"] = 'BulletDefault'] = "BulletDefault";
        TemplateEnum[TemplateEnum["BulletDisk"] = 'BulletDisk'] = "BulletDisk";
        TemplateEnum[TemplateEnum["BulletCircle"] = 'BulletCircle'] = "BulletCircle";
        TemplateEnum[TemplateEnum["BulletSquare"] = 'BulletSquare'] = "BulletSquare";
        TemplateEnum[TemplateEnum["BulletDiamonds"] = 'BulletDiamonds'] = "BulletDiamonds";
        TemplateEnum[TemplateEnum["BulletArrowHead"] = 'BulletArrowHead'] = "BulletArrowHead";
        TemplateEnum[TemplateEnum["BulletTick"] = 'BulletTick'] = "BulletTick";
        TemplateEnum[TemplateEnum["NumberDefault"] = 'NumberDefault'] = "NumberDefault";
        TemplateEnum[TemplateEnum["NumberArabicDot"] = 'NumberArabicDot'] = "NumberArabicDot";
        TemplateEnum[TemplateEnum["NumberArabicParenthesis"] = 'NumberArabicParenthesis'] = "NumberArabicParenthesis";
        TemplateEnum[TemplateEnum["NumberUppercaseRomanDot"] = 'NumberUppercaseRomanDot'] = "NumberUppercaseRomanDot";
        TemplateEnum[TemplateEnum["NumberUppercaseLetterDot"] = 'NumberUppercaseLetterDot'] = "NumberUppercaseLetterDot";
        TemplateEnum[TemplateEnum["NumberLowercaseLetterParenthesis"] = 'NumberLowercaseLetterParenthesis'] = "NumberLowercaseLetterParenthesis";
        TemplateEnum[TemplateEnum["NumberLowercaseLetterDot"] = 'NumberLowercaseLetterDot'] = "NumberLowercaseLetterDot";
        TemplateEnum[TemplateEnum["NumberLowercaseRomanDot"] = 'NumberLowercaseRomanDot'] = "NumberLowercaseRomanDot";
        TemplateEnum[TemplateEnum["OutlineNumbers"] = 'OutlineNumbers'] = "OutlineNumbers";
        TemplateEnum[TemplateEnum["OutlineLegal"] = 'OutlineLegal'] = "OutlineLegal";
        TemplateEnum[TemplateEnum["OutlineBullets"] = 'OutlineBullets'] = "OutlineBullets";
        TemplateEnum[TemplateEnum["OutlineHeadingsArticleSection"] = 'OutlineHeadingsArticleSection'] = "OutlineHeadingsArticleSection";
        TemplateEnum[TemplateEnum["OutlineHeadingsLegal"] = 'OutlineHeadingsLegal'] = "OutlineHeadingsLegal";
        TemplateEnum[TemplateEnum["OutlineHeadingsNumbers"] = 'OutlineHeadingsNumbers'] = "OutlineHeadingsNumbers";
        TemplateEnum[TemplateEnum["OutlineHeadingsChapter"] = 'OutlineHeadingsChapter'] = "OutlineHeadingsChapter";
    })(TemplateEnum = ListInsert.TemplateEnum || (ListInsert.TemplateEnum = {}));
})(ListInsert = exports.ListInsert || (exports.ListInsert = {}));
// tslint:enable:quotemark
