// asposewordscloud/dist/model/metafileRenderingOptionsData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.importsMapMetafileRenderingOptionsData = {};
/**
 * container class for options of metafile rendering.
 */
class MetafileRenderingOptionsData {
    constructor(init) {
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return MetafileRenderingOptionsData.attributeTypeMap;
    }
}
exports.MetafileRenderingOptionsData = MetafileRenderingOptionsData;
/**
 * Attribute type map
 */
MetafileRenderingOptionsData.attributeTypeMap = [
    {
        name: "emfPlusDualRenderingMode",
        baseName: "EmfPlusDualRenderingMode",
        type: "string",
    },
    {
        name: "emulateRasterOperations",
        baseName: "EmulateRasterOperations",
        type: "boolean",
    },
    {
        name: "renderingMode",
        baseName: "RenderingMode",
        type: "string",
    },
    {
        name: "useEmfEmbeddedToWmf",
        baseName: "UseEmfEmbeddedToWmf",
        type: "boolean",
    },
    {
        name: "scaleWmfFontsToMetafileSize",
        baseName: "ScaleWmfFontsToMetafileSize",
        type: "boolean",
    }
];
