// asposewordscloud/dist/model/officeMathObjectsResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const officeMathObjectsCollection_1 = require("./officeMathObjectsCollection");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapOfficeMathObjectsResponse = {
    OfficeMathObjectsCollection: officeMathObjectsCollection_1.OfficeMathObjectsCollection,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/OfficeMathObjects.
 */
class OfficeMathObjectsResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(OfficeMathObjectsResponse.attributeTypeMap);
    }
}
exports.OfficeMathObjectsResponse = OfficeMathObjectsResponse;
/**
 * Attribute type map
 */
OfficeMathObjectsResponse.attributeTypeMap = [
    {
        name: "officeMathObjects",
        baseName: "OfficeMathObjects",
        type: "OfficeMathObjectsCollection",
    }
];
