// asposewordscloud/dist/model/officeMathObjectsCollection.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const linkElement_1 = require("./linkElement");
const officeMathObject_1 = require("./officeMathObject");
const wordsApiLink_1 = require("./wordsApiLink");
exports.importsMapOfficeMathObjectsCollection = {
    LinkElement: linkElement_1.LinkElement,
    OfficeMathObject: officeMathObject_1.OfficeMathObject,
    WordsApiLink: wordsApiLink_1.WordsApiLink,
};
/**
 * Collection of OfficeMath objects.
 */
class OfficeMathObjectsCollection extends linkElement_1.LinkElement {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(OfficeMathObjectsCollection.attributeTypeMap);
    }
}
exports.OfficeMathObjectsCollection = OfficeMathObjectsCollection;
/**
 * Attribute type map
 */
OfficeMathObjectsCollection.attributeTypeMap = [
    {
        name: "list",
        baseName: "List",
        type: "Array<OfficeMathObject>",
    }
];
