// asposewordscloud/dist/model/officeMathObjectResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const officeMathObject_1 = require("./officeMathObject");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapOfficeMathObjectResponse = {
    OfficeMathObject: officeMathObject_1.OfficeMathObject,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/officeMathObjects/0.
 */
class OfficeMathObjectResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(OfficeMathObjectResponse.attributeTypeMap);
    }
}
exports.OfficeMathObjectResponse = OfficeMathObjectResponse;
/**
 * Attribute type map
 */
OfficeMathObjectResponse.attributeTypeMap = [
    {
        name: "officeMathObject",
        baseName: "OfficeMathObject",
        type: "OfficeMathObject",
    }
];
