// asposewordscloud/dist/model/outlineOptionsData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const bookmarksOutlineLevelData_1 = require("./bookmarksOutlineLevelData");
exports.importsMapOutlineOptionsData = {
    BookmarksOutlineLevelData: bookmarksOutlineLevelData_1.BookmarksOutlineLevelData,
};
/**
 * container class for outline options.
 */
class OutlineOptionsData {
    constructor(init) {
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return OutlineOptionsData.attributeTypeMap;
    }
}
exports.OutlineOptionsData = OutlineOptionsData;
/**
 * Attribute type map
 */
OutlineOptionsData.attributeTypeMap = [
    {
        name: "bookmarksOutlineLevels",
        baseName: "BookmarksOutlineLevels",
        type: "Array<BookmarksOutlineLevelData>",
    },
    {
        name: "defaultBookmarksOutlineLevel",
        baseName: "DefaultBookmarksOutlineLevel",
        type: "number",
    },
    {
        name: "createMissingOutlineLevels",
        baseName: "CreateMissingOutlineLevels",
        type: "boolean",
    },
    {
        name: "createOutlinesForHeadingsInTables",
        baseName: "CreateOutlinesForHeadingsInTables",
        type: "boolean",
    },
    {
        name: "expandedOutlineLevels",
        baseName: "ExpandedOutlineLevels",
        type: "number",
    },
    {
        name: "headingsOutlineLevels",
        baseName: "HeadingsOutlineLevels",
        type: "number",
    }
];
