// asposewordscloud/dist/model/pageStatData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const footnotesStatData_1 = require("./footnotesStatData");
exports.importsMapPageStatData = {
    FootnotesStatData: footnotesStatData_1.FootnotesStatData,
};
/**
 * Container for the page's statistical data.
 */
class PageStatData {
    constructor(init) {
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return PageStatData.attributeTypeMap;
    }
}
exports.PageStatData = PageStatData;
/**
 * Attribute type map
 */
PageStatData.attributeTypeMap = [
    {
        name: "pageNumber",
        baseName: "PageNumber",
        type: "number",
    },
    {
        name: "wordCount",
        baseName: "WordCount",
        type: "number",
    },
    {
        name: "paragraphCount",
        baseName: "ParagraphCount",
        type: "number",
    },
    {
        name: "footnotesStatData",
        baseName: "FootnotesStatData",
        type: "FootnotesStatData",
    }
];
