// asposewordscloud/dist/model/paragraphLinkCollectionResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const paragraphLinkCollection_1 = require("./paragraphLinkCollection");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapParagraphLinkCollectionResponse = {
    ParagraphLinkCollection: paragraphLinkCollection_1.ParagraphLinkCollection,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/paragraphs.
 */
class ParagraphLinkCollectionResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(ParagraphLinkCollectionResponse.attributeTypeMap);
    }
}
exports.ParagraphLinkCollectionResponse = ParagraphLinkCollectionResponse;
/**
 * Attribute type map
 */
ParagraphLinkCollectionResponse.attributeTypeMap = [
    {
        name: "paragraphs",
        baseName: "Paragraphs",
        type: "ParagraphLinkCollection",
    }
];
