// asposewordscloud/dist/model/paragraphListFormatResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const listFormat_1 = require("./listFormat");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapParagraphListFormatResponse = {
    ListFormat: listFormat_1.ListFormat,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/paragraphs/{0}/listFormat.
 */
class ParagraphListFormatResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(ParagraphListFormatResponse.attributeTypeMap);
    }
}
exports.ParagraphListFormatResponse = ParagraphListFormatResponse;
/**
 * Attribute type map
 */
ParagraphListFormatResponse.attributeTypeMap = [
    {
        name: "listFormat",
        baseName: "ListFormat",
        type: "ListFormat",
    }
];
