// asposewordscloud/dist/model/paragraphFormatResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const paragraphFormat_1 = require("./paragraphFormat");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapParagraphFormatResponse = {
    ParagraphFormat: paragraphFormat_1.ParagraphFormat,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/paragraphs/{0}/format.
 */
class ParagraphFormatResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(ParagraphFormatResponse.attributeTypeMap);
    }
}
exports.ParagraphFormatResponse = ParagraphFormatResponse;
/**
 * Attribute type map
 */
ParagraphFormatResponse.attributeTypeMap = [
    {
        name: "paragraphFormat",
        baseName: "ParagraphFormat",
        type: "ParagraphFormat",
    }
];
