// asposewordscloud/dist/model/paragraphLinkCollection.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const linkElement_1 = require("./linkElement");
const paragraphLink_1 = require("./paragraphLink");
const wordsApiLink_1 = require("./wordsApiLink");
exports.importsMapParagraphLinkCollection = {
    LinkElement: linkElement_1.LinkElement,
    ParagraphLink: paragraphLink_1.ParagraphLink,
    WordsApiLink: wordsApiLink_1.WordsApiLink,
};
/**
 * Collection of links to paragraphs.
 */
class ParagraphLinkCollection extends linkElement_1.LinkElement {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(ParagraphLinkCollection.attributeTypeMap);
    }
}
exports.ParagraphLinkCollection = ParagraphLinkCollection;
/**
 * Attribute type map
 */
ParagraphLinkCollection.attributeTypeMap = [
    {
        name: "paragraphLinkList",
        baseName: "ParagraphLinkList",
        type: "Array<ParagraphLink>",
    }
];
