// asposewordscloud/dist/model/pclSaveOptionsData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const fixedPageSaveOptionsData_1 = require("./fixedPageSaveOptionsData");
const metafileRenderingOptionsData_1 = require("./metafileRenderingOptionsData");
exports.importsMapPclSaveOptionsData = {
    FixedPageSaveOptionsData: fixedPageSaveOptionsData_1.FixedPageSaveOptionsData,
    MetafileRenderingOptionsData: metafileRenderingOptionsData_1.MetafileRenderingOptionsData,
};
/**
 * Container class for pcl save options.
 */
class PclSaveOptionsData extends fixedPageSaveOptionsData_1.FixedPageSaveOptionsData {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(PclSaveOptionsData.attributeTypeMap);
    }
}
exports.PclSaveOptionsData = PclSaveOptionsData;
/**
 * Attribute type map
 */
PclSaveOptionsData.attributeTypeMap = [
    {
        name: "falllbackFontName",
        baseName: "FalllbackFontName",
        type: "string",
    },
    {
        name: "rasterizeTransformedElements",
        baseName: "RasterizeTransformedElements",
        type: "boolean",
    }
];
