// asposewordscloud/dist/model/pdfDigitalSignatureDetailsData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.importsMapPdfDigitalSignatureDetailsData = {};
/**
 * container class for details of digital signature.
 */
class PdfDigitalSignatureDetailsData {
    constructor(init) {
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return PdfDigitalSignatureDetailsData.attributeTypeMap;
    }
}
exports.PdfDigitalSignatureDetailsData = PdfDigitalSignatureDetailsData;
/**
 * Attribute type map
 */
PdfDigitalSignatureDetailsData.attributeTypeMap = [
    {
        name: "certificateFilename",
        baseName: "CertificateFilename",
        type: "string",
    },
    {
        name: "hashAlgorithm",
        baseName: "HashAlgorithm",
        type: "string",
    },
    {
        name: "location",
        baseName: "Location",
        type: "string",
    },
    {
        name: "reason",
        baseName: "Reason",
        type: "string",
    },
    {
        name: "signatureDate",
        baseName: "SignatureDate",
        type: "Date",
    }
];
