// asposewordscloud/dist/model/preferredWidth.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.importsMapPreferredWidth = {};
/**
 * Preferred width.
 */
class PreferredWidth {
    constructor(init) {
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return PreferredWidth.attributeTypeMap;
    }
}
exports.PreferredWidth = PreferredWidth;
/**
 * Attribute type map
 */
PreferredWidth.attributeTypeMap = [
    {
        name: "type",
        baseName: "Type",
        type: "PreferredWidth.TypeEnum",
    },
    {
        name: "value",
        baseName: "Value",
        type: "number",
    }
];
/**
 * Enums for PreferredWidth
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (PreferredWidth) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Auto"] = 'Auto'] = "Auto";
        TypeEnum[TypeEnum["Percent"] = 'Percent'] = "Percent";
        TypeEnum[TypeEnum["Points"] = 'Points'] = "Points";
    })(TypeEnum = PreferredWidth.TypeEnum || (PreferredWidth.TypeEnum = {}));
})(PreferredWidth = exports.PreferredWidth || (exports.PreferredWidth = {}));
// tslint:enable:quotemark
