// asposewordscloud/dist/model/protectionDataResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const fileLink_1 = require("./fileLink");
const protectionData_1 = require("./protectionData");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapProtectionDataResponse = {
    FileLink: fileLink_1.FileLink,
    ProtectionData: protectionData_1.ProtectionData,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * Response for the request of data about protection.
 */
class ProtectionDataResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(ProtectionDataResponse.attributeTypeMap);
    }
}
exports.ProtectionDataResponse = ProtectionDataResponse;
/**
 * Attribute type map
 */
ProtectionDataResponse.attributeTypeMap = [
    {
        name: "documentLink",
        baseName: "DocumentLink",
        type: "FileLink",
    },
    {
        name: "protectionData",
        baseName: "ProtectionData",
        type: "ProtectionData",
    }
];
