// asposewordscloud/dist/model/psSaveOptionsData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const fixedPageSaveOptionsData_1 = require("./fixedPageSaveOptionsData");
const metafileRenderingOptionsData_1 = require("./metafileRenderingOptionsData");
exports.importsMapPsSaveOptionsData = {
    FixedPageSaveOptionsData: fixedPageSaveOptionsData_1.FixedPageSaveOptionsData,
    MetafileRenderingOptionsData: metafileRenderingOptionsData_1.MetafileRenderingOptionsData,
};
/**
 * container class for ps save options.
 */
class PsSaveOptionsData extends fixedPageSaveOptionsData_1.FixedPageSaveOptionsData {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(PsSaveOptionsData.attributeTypeMap);
    }
}
exports.PsSaveOptionsData = PsSaveOptionsData;
/**
 * Attribute type map
 */
PsSaveOptionsData.attributeTypeMap = [
    {
        name: "useBookFoldPrintingSettings",
        baseName: "UseBookFoldPrintingSettings",
        type: "boolean",
    }
];
