// asposewordscloud/dist/model/replaceTextResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const fileLink_1 = require("./fileLink");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapReplaceTextResponse = {
    FileLink: fileLink_1.FileLink,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * Response for \"Replace text\" action.
 */
class ReplaceTextResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(ReplaceTextResponse.attributeTypeMap);
    }
}
exports.ReplaceTextResponse = ReplaceTextResponse;
/**
 * Attribute type map
 */
ReplaceTextResponse.attributeTypeMap = [
    {
        name: "documentLink",
        baseName: "DocumentLink",
        type: "FileLink",
    },
    {
        name: "matches",
        baseName: "Matches",
        type: "number",
    }
];
