// asposewordscloud/dist/model/replaceTextParameters.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.importsMapReplaceTextParameters = {};
/**
 * Class for document replace text request building.
 */
class ReplaceTextParameters {
    constructor(init) {
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return ReplaceTextParameters.attributeTypeMap;
    }
}
exports.ReplaceTextParameters = ReplaceTextParameters;
/**
 * Attribute type map
 */
ReplaceTextParameters.attributeTypeMap = [
    {
        name: "oldValue",
        baseName: "OldValue",
        type: "string",
    },
    {
        name: "newValue",
        baseName: "NewValue",
        type: "string",
    },
    {
        name: "isMatchCase",
        baseName: "IsMatchCase",
        type: "boolean",
    },
    {
        name: "isMatchWholeWord",
        baseName: "IsMatchWholeWord",
        type: "boolean",
    },
    {
        name: "isOldValueRegex",
        baseName: "IsOldValueRegex",
        type: "boolean",
    }
];
