// asposewordscloud/dist/model/revisionsModificationResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const modificationOperationResult_1 = require("./modificationOperationResult");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapRevisionsModificationResponse = {
    ModificationOperationResult: modificationOperationResult_1.ModificationOperationResult,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * response of the modification operations for the revisions collection (now these are acceptAll and rejectAll).
 */
class RevisionsModificationResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(RevisionsModificationResponse.attributeTypeMap);
    }
}
exports.RevisionsModificationResponse = RevisionsModificationResponse;
/**
 * Attribute type map
 */
RevisionsModificationResponse.attributeTypeMap = [
    {
        name: "result",
        baseName: "Result",
        type: "ModificationOperationResult",
    }
];
