// asposewordscloud/dist/model/rtfSaveOptionsData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const saveOptionsData_1 = require("./saveOptionsData");
exports.importsMapRtfSaveOptionsData = {
    SaveOptionsData: saveOptionsData_1.SaveOptionsData,
};
/**
 * container class for rtf save options.
 */
class RtfSaveOptionsData extends saveOptionsData_1.SaveOptionsData {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(RtfSaveOptionsData.attributeTypeMap);
    }
}
exports.RtfSaveOptionsData = RtfSaveOptionsData;
/**
 * Attribute type map
 */
RtfSaveOptionsData.attributeTypeMap = [
    {
        name: "exportCompactSize",
        baseName: "ExportCompactSize",
        type: "boolean",
    },
    {
        name: "exportImagesForOldReaders",
        baseName: "ExportImagesForOldReaders",
        type: "boolean",
    },
    {
        name: "prettyFormat",
        baseName: "PrettyFormat",
        type: "boolean",
    },
    {
        name: "saveImagesAsWmf",
        baseName: "SaveImagesAsWmf",
        type: "boolean",
    }
];
// tslint:enable:quotemark
