// asposewordscloud/dist/model/saveOptionsData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.importsMapSaveOptionsData = {};
/**
 * base container class for save options data.
 */
class SaveOptionsData {
    constructor(init) {
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return SaveOptionsData.attributeTypeMap;
    }
}
exports.SaveOptionsData = SaveOptionsData;
/**
 * Attribute type map
 */
SaveOptionsData.attributeTypeMap = [
    {
        name: "saveFormat",
        baseName: "SaveFormat",
        type: "string",
    },
    {
        name: "fileName",
        baseName: "FileName",
        type: "string",
    },
    {
        name: "dmlRenderingMode",
        baseName: "DmlRenderingMode",
        type: "string",
    },
    {
        name: "dmlEffectsRenderingMode",
        baseName: "DmlEffectsRenderingMode",
        type: "string",
    },
    {
        name: "zipOutput",
        baseName: "ZipOutput",
        type: "boolean",
    },
    {
        name: "updateLastSavedTimeProperty",
        baseName: "UpdateLastSavedTimeProperty",
        type: "boolean",
    },
    {
        name: "updateSdtContent",
        baseName: "UpdateSdtContent",
        type: "boolean",
    },
    {
        name: "updateFields",
        baseName: "UpdateFields",
        type: "boolean",
    },
    {
        name: "dml3DEffectsRenderingMode",
        baseName: "Dml3DEffectsRenderingMode",
        type: "SaveOptionsData.Dml3DEffectsRenderingModeEnum",
    },
    {
        name: "updateLastPrintedProperty",
        baseName: "UpdateLastPrintedProperty",
        type: "boolean",
    }
];
/**
 * Enums for SaveOptionsData
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (SaveOptionsData) {
    let Dml3DEffectsRenderingModeEnum;
    (function (Dml3DEffectsRenderingModeEnum) {
        Dml3DEffectsRenderingModeEnum[Dml3DEffectsRenderingModeEnum["Basic"] = 'Basic'] = "Basic";
        Dml3DEffectsRenderingModeEnum[Dml3DEffectsRenderingModeEnum["Advanced"] = 'Advanced'] = "Advanced";
    })(Dml3DEffectsRenderingModeEnum = SaveOptionsData.Dml3DEffectsRenderingModeEnum || (SaveOptionsData.Dml3DEffectsRenderingModeEnum = {}));
})(SaveOptionsData = exports.SaveOptionsData || (exports.SaveOptionsData = {}));
// tslint:enable:quotemark
