// asposewordscloud/dist/model/searchResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const searchResultsCollection_1 = require("./searchResultsCollection");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapSearchResponse = {
    SearchResultsCollection: searchResultsCollection_1.SearchResultsCollection,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/search.
 */
class SearchResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(SearchResponse.attributeTypeMap);
    }
}
exports.SearchResponse = SearchResponse;
/**
 * Attribute type map
 */
SearchResponse.attributeTypeMap = [
    {
        name: "searchingPattern",
        baseName: "SearchingPattern",
        type: "string",
    },
    {
        name: "searchResults",
        baseName: "SearchResults",
        type: "SearchResultsCollection",
    }
];
