// asposewordscloud/dist/model/searchResultsCollection.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const linkElement_1 = require("./linkElement");
const searchResult_1 = require("./searchResult");
const wordsApiLink_1 = require("./wordsApiLink");
exports.importsMapSearchResultsCollection = {
    LinkElement: linkElement_1.LinkElement,
    SearchResult: searchResult_1.SearchResult,
    WordsApiLink: wordsApiLink_1.WordsApiLink,
};
/**
 * Collection of search results.
 */
class SearchResultsCollection extends linkElement_1.LinkElement {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(SearchResultsCollection.attributeTypeMap);
    }
}
exports.SearchResultsCollection = SearchResultsCollection;
/**
 * Attribute type map
 */
SearchResultsCollection.attributeTypeMap = [
    {
        name: "resultsList",
        baseName: "ResultsList",
        type: "Array<SearchResult>",
    }
];
