// asposewordscloud/dist/model/sectionPageSetupResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const pageSetup_1 = require("./pageSetup");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapSectionPageSetupResponse = {
    PageSetup: pageSetup_1.PageSetup,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/sections/{0}/PageSetup.
 */
class SectionPageSetupResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(SectionPageSetupResponse.attributeTypeMap);
    }
}
exports.SectionPageSetupResponse = SectionPageSetupResponse;
/**
 * Attribute type map
 */
SectionPageSetupResponse.attributeTypeMap = [
    {
        name: "pageSetup",
        baseName: "PageSetup",
        type: "PageSetup",
    }
];
