// asposewordscloud/dist/model/section.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const linkElement_1 = require("./linkElement");
const nodeLink_1 = require("./nodeLink");
const wordsApiLink_1 = require("./wordsApiLink");
exports.importsMapSection = {
    LinkElement: linkElement_1.LinkElement,
    NodeLink: nodeLink_1.NodeLink,
    WordsApiLink: wordsApiLink_1.WordsApiLink,
};
/**
 * Section element.
 */
class Section extends linkElement_1.LinkElement {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(Section.attributeTypeMap);
    }
}
exports.Section = Section;
/**
 * Attribute type map
 */
Section.attributeTypeMap = [
    {
        name: "childNodes",
        baseName: "ChildNodes",
        type: "Array<NodeLink>",
    },
    {
        name: "headerFooters",
        baseName: "HeaderFooters",
        type: "LinkElement",
    },
    {
        name: "pageSetup",
        baseName: "PageSetup",
        type: "LinkElement",
    },
    {
        name: "paragraphs",
        baseName: "Paragraphs",
        type: "LinkElement",
    },
    {
        name: "tables",
        baseName: "Tables",
        type: "LinkElement",
    }
];
