// asposewordscloud/dist/model/sectionLinkCollectionResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const sectionLinkCollection_1 = require("./sectionLinkCollection");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapSectionLinkCollectionResponse = {
    SectionLinkCollection: sectionLinkCollection_1.SectionLinkCollection,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/sections.
 */
class SectionLinkCollectionResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(SectionLinkCollectionResponse.attributeTypeMap);
    }
}
exports.SectionLinkCollectionResponse = SectionLinkCollectionResponse;
/**
 * Attribute type map
 */
SectionLinkCollectionResponse.attributeTypeMap = [
    {
        name: "sections",
        baseName: "Sections",
        type: "SectionLinkCollection",
    }
];
