// asposewordscloud/dist/model/sectionLinkCollection.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const linkElement_1 = require("./linkElement");
const sectionLink_1 = require("./sectionLink");
const wordsApiLink_1 = require("./wordsApiLink");
exports.importsMapSectionLinkCollection = {
    LinkElement: linkElement_1.LinkElement,
    SectionLink: sectionLink_1.SectionLink,
    WordsApiLink: wordsApiLink_1.WordsApiLink,
};
/**
 * Collection of links to sections.
 */
class SectionLinkCollection extends linkElement_1.LinkElement {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(SectionLinkCollection.attributeTypeMap);
    }
}
exports.SectionLinkCollection = SectionLinkCollection;
/**
 * Attribute type map
 */
SectionLinkCollection.attributeTypeMap = [
    {
        name: "sectionLinkList",
        baseName: "SectionLinkList",
        type: "Array<SectionLink>",
    }
];
