// asposewordscloud/dist/model/shading.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const xmlColor_1 = require("./xmlColor");
exports.importsMapShading = {
    XmlColor: xmlColor_1.XmlColor,
};
/**
 * Paragraph format shading element.
 */
class Shading {
    constructor(init) {
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return Shading.attributeTypeMap;
    }
}
exports.Shading = Shading;
/**
 * Attribute type map
 */
Shading.attributeTypeMap = [
    {
        name: "backgroundPatternColor",
        baseName: "BackgroundPatternColor",
        type: "XmlColor",
    },
    {
        name: "foregroundPatternColor",
        baseName: "ForegroundPatternColor",
        type: "XmlColor",
    },
    {
        name: "texture",
        baseName: "Texture",
        type: "Shading.TextureEnum",
    }
];
/**
 * Enums for Shading
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (Shading) {
    let TextureEnum;
    (function (TextureEnum) {
        TextureEnum[TextureEnum["TextureNone"] = 'TextureNone'] = "TextureNone";
        TextureEnum[TextureEnum["TextureSolid"] = 'TextureSolid'] = "TextureSolid";
        TextureEnum[TextureEnum["Texture5Percent"] = 'Texture5Percent'] = "Texture5Percent";
        TextureEnum[TextureEnum["Texture10Percent"] = 'Texture10Percent'] = "Texture10Percent";
        TextureEnum[TextureEnum["Texture20Percent"] = 'Texture20Percent'] = "Texture20Percent";
        TextureEnum[TextureEnum["Texture25Percent"] = 'Texture25Percent'] = "Texture25Percent";
        TextureEnum[TextureEnum["Texture30Percent"] = 'Texture30Percent'] = "Texture30Percent";
        TextureEnum[TextureEnum["Texture40Percent"] = 'Texture40Percent'] = "Texture40Percent";
        TextureEnum[TextureEnum["Texture50Percent"] = 'Texture50Percent'] = "Texture50Percent";
        TextureEnum[TextureEnum["Texture60Percent"] = 'Texture60Percent'] = "Texture60Percent";
        TextureEnum[TextureEnum["Texture70Percent"] = 'Texture70Percent'] = "Texture70Percent";
        TextureEnum[TextureEnum["Texture75Percent"] = 'Texture75Percent'] = "Texture75Percent";
        TextureEnum[TextureEnum["Texture80Percent"] = 'Texture80Percent'] = "Texture80Percent";
        TextureEnum[TextureEnum["Texture90Percent"] = 'Texture90Percent'] = "Texture90Percent";
        TextureEnum[TextureEnum["TextureDarkHorizontal"] = 'TextureDarkHorizontal'] = "TextureDarkHorizontal";
        TextureEnum[TextureEnum["TextureDarkVertical"] = 'TextureDarkVertical'] = "TextureDarkVertical";
        TextureEnum[TextureEnum["TextureDarkDiagonalDown"] = 'TextureDarkDiagonalDown'] = "TextureDarkDiagonalDown";
        TextureEnum[TextureEnum["TextureDarkDiagonalUp"] = 'TextureDarkDiagonalUp'] = "TextureDarkDiagonalUp";
        TextureEnum[TextureEnum["TextureDarkCross"] = 'TextureDarkCross'] = "TextureDarkCross";
        TextureEnum[TextureEnum["TextureDarkDiagonalCross"] = 'TextureDarkDiagonalCross'] = "TextureDarkDiagonalCross";
        TextureEnum[TextureEnum["TextureHorizontal"] = 'TextureHorizontal'] = "TextureHorizontal";
        TextureEnum[TextureEnum["TextureVertical"] = 'TextureVertical'] = "TextureVertical";
        TextureEnum[TextureEnum["TextureDiagonalDown"] = 'TextureDiagonalDown'] = "TextureDiagonalDown";
        TextureEnum[TextureEnum["TextureDiagonalUp"] = 'TextureDiagonalUp'] = "TextureDiagonalUp";
        TextureEnum[TextureEnum["TextureCross"] = 'TextureCross'] = "TextureCross";
        TextureEnum[TextureEnum["TextureDiagonalCross"] = 'TextureDiagonalCross'] = "TextureDiagonalCross";
        TextureEnum[TextureEnum["Texture2Pt5Percent"] = 'Texture2Pt5Percent'] = "Texture2Pt5Percent";
        TextureEnum[TextureEnum["Texture7Pt5Percent"] = 'Texture7Pt5Percent'] = "Texture7Pt5Percent";
        TextureEnum[TextureEnum["Texture12Pt5Percent"] = 'Texture12Pt5Percent'] = "Texture12Pt5Percent";
        TextureEnum[TextureEnum["Texture15Percent"] = 'Texture15Percent'] = "Texture15Percent";
        TextureEnum[TextureEnum["Texture17Pt5Percent"] = 'Texture17Pt5Percent'] = "Texture17Pt5Percent";
        TextureEnum[TextureEnum["Texture22Pt5Percent"] = 'Texture22Pt5Percent'] = "Texture22Pt5Percent";
        TextureEnum[TextureEnum["Texture27Pt5Percent"] = 'Texture27Pt5Percent'] = "Texture27Pt5Percent";
        TextureEnum[TextureEnum["Texture32Pt5Percent"] = 'Texture32Pt5Percent'] = "Texture32Pt5Percent";
        TextureEnum[TextureEnum["Texture35Percent"] = 'Texture35Percent'] = "Texture35Percent";
        TextureEnum[TextureEnum["Texture37Pt5Percent"] = 'Texture37Pt5Percent'] = "Texture37Pt5Percent";
        TextureEnum[TextureEnum["Texture42Pt5Percent"] = 'Texture42Pt5Percent'] = "Texture42Pt5Percent";
        TextureEnum[TextureEnum["Texture45Percent"] = 'Texture45Percent'] = "Texture45Percent";
        TextureEnum[TextureEnum["Texture47Pt5Percent"] = 'Texture47Pt5Percent'] = "Texture47Pt5Percent";
        TextureEnum[TextureEnum["Texture52Pt5Percent"] = 'Texture52Pt5Percent'] = "Texture52Pt5Percent";
        TextureEnum[TextureEnum["Texture55Percent"] = 'Texture55Percent'] = "Texture55Percent";
        TextureEnum[TextureEnum["Texture57Pt5Percent"] = 'Texture57Pt5Percent'] = "Texture57Pt5Percent";
        TextureEnum[TextureEnum["Texture62Pt5Percent"] = 'Texture62Pt5Percent'] = "Texture62Pt5Percent";
        TextureEnum[TextureEnum["Texture65Percent"] = 'Texture65Percent'] = "Texture65Percent";
        TextureEnum[TextureEnum["Texture67Pt5Percent"] = 'Texture67Pt5Percent'] = "Texture67Pt5Percent";
        TextureEnum[TextureEnum["Texture72Pt5Percent"] = 'Texture72Pt5Percent'] = "Texture72Pt5Percent";
        TextureEnum[TextureEnum["Texture77Pt5Percent"] = 'Texture77Pt5Percent'] = "Texture77Pt5Percent";
        TextureEnum[TextureEnum["Texture82Pt5Percent"] = 'Texture82Pt5Percent'] = "Texture82Pt5Percent";
        TextureEnum[TextureEnum["Texture85Percent"] = 'Texture85Percent'] = "Texture85Percent";
        TextureEnum[TextureEnum["Texture87Pt5Percent"] = 'Texture87Pt5Percent'] = "Texture87Pt5Percent";
        TextureEnum[TextureEnum["Texture92Pt5Percent"] = 'Texture92Pt5Percent'] = "Texture92Pt5Percent";
        TextureEnum[TextureEnum["Texture95Percent"] = 'Texture95Percent'] = "Texture95Percent";
        TextureEnum[TextureEnum["Texture97Pt5Percent"] = 'Texture97Pt5Percent'] = "Texture97Pt5Percent";
        TextureEnum[TextureEnum["TextureNil"] = 'TextureNil'] = "TextureNil";
    })(TextureEnum = Shading.TextureEnum || (Shading.TextureEnum = {}));
})(Shading = exports.Shading || (exports.Shading = {}));
// tslint:enable:quotemark
