// asposewordscloud/dist/model/splitDocumentResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const splitDocumentResult_1 = require("./splitDocumentResult");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapSplitDocumentResponse = {
    SplitDocumentResult: splitDocumentResult_1.SplitDocumentResult,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling:  POST /{name}/split .
 */
class SplitDocumentResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(SplitDocumentResponse.attributeTypeMap);
    }
}
exports.SplitDocumentResponse = SplitDocumentResponse;
/**
 * Attribute type map
 */
SplitDocumentResponse.attributeTypeMap = [
    {
        name: "splitResult",
        baseName: "SplitResult",
        type: "SplitDocumentResult",
    }
];
