// asposewordscloud/dist/model/statDataResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const documentStatData_1 = require("./documentStatData");
const fileLink_1 = require("./fileLink");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapStatDataResponse = {
    DocumentStatData: documentStatData_1.DocumentStatData,
    FileLink: fileLink_1.FileLink,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * Response for the request of the document's statistical data.
 */
class StatDataResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(StatDataResponse.attributeTypeMap);
    }
}
exports.StatDataResponse = StatDataResponse;
/**
 * Attribute type map
 */
StatDataResponse.attributeTypeMap = [
    {
        name: "documentLink",
        baseName: "DocumentLink",
        type: "FileLink",
    },
    {
        name: "statData",
        baseName: "StatData",
        type: "DocumentStatData",
    }
];
