// asposewordscloud/dist/model/styleInsert.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.importsMapStyleInsert = {};
/**
 * Represents a single document style to insert.
 */
class StyleInsert {
    constructor(init) {
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return StyleInsert.attributeTypeMap;
    }
}
exports.StyleInsert = StyleInsert;
/**
 * Attribute type map
 */
StyleInsert.attributeTypeMap = [
    {
        name: "styleName",
        baseName: "StyleName",
        type: "string",
    },
    {
        name: "styleType",
        baseName: "StyleType",
        type: "StyleInsert.StyleTypeEnum",
    }
];
/**
 * Enums for StyleInsert
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (StyleInsert) {
    let StyleTypeEnum;
    (function (StyleTypeEnum) {
        StyleTypeEnum[StyleTypeEnum["Paragraph"] = 'Paragraph'] = "Paragraph";
        StyleTypeEnum[StyleTypeEnum["Character"] = 'Character'] = "Character";
        StyleTypeEnum[StyleTypeEnum["Table"] = 'Table'] = "Table";
        StyleTypeEnum[StyleTypeEnum["List"] = 'List'] = "List";
    })(StyleTypeEnum = StyleInsert.StyleTypeEnum || (StyleInsert.StyleTypeEnum = {}));
})(StyleInsert = exports.StyleInsert || (exports.StyleInsert = {}));
// tslint:enable:quotemark
