// asposewordscloud/dist/model/tableCellFormatResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tableCellFormat_1 = require("./tableCellFormat");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapTableCellFormatResponse = {
    TableCellFormat: tableCellFormat_1.TableCellFormat,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/tables/{0}/rows/{1}/cells/{2}/cellformat.
 */
class TableCellFormatResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(TableCellFormatResponse.attributeTypeMap);
    }
}
exports.TableCellFormatResponse = TableCellFormatResponse;
/**
 * Attribute type map
 */
TableCellFormatResponse.attributeTypeMap = [
    {
        name: "cellFormat",
        baseName: "CellFormat",
        type: "TableCellFormat",
    }
];
