// asposewordscloud/dist/model/table.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const nodeLink_1 = require("./nodeLink");
const tableProperties_1 = require("./tableProperties");
const tableRow_1 = require("./tableRow");
exports.importsMapTable = {
    NodeLink: nodeLink_1.NodeLink,
    TableProperties: tableProperties_1.TableProperties,
    TableRow: tableRow_1.TableRow,
};
/**
 * Table element.
 */
class Table extends nodeLink_1.NodeLink {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(Table.attributeTypeMap);
    }
}
exports.Table = Table;
/**
 * Attribute type map
 */
Table.attributeTypeMap = [
    {
        name: "tableProperties",
        baseName: "TableProperties",
        type: "TableProperties",
    },
    {
        name: "tableRowList",
        baseName: "TableRowList",
        type: "Array<TableRow>",
    }
];
