// asposewordscloud/dist/model/tableRowFormatResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tableRowFormat_1 = require("./tableRowFormat");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapTableRowFormatResponse = {
    TableRowFormat: tableRowFormat_1.TableRowFormat,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/tables/{0}/rows/{1}/rowformat.
 */
class TableRowFormatResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(TableRowFormatResponse.attributeTypeMap);
    }
}
exports.TableRowFormatResponse = TableRowFormatResponse;
/**
 * Attribute type map
 */
TableRowFormatResponse.attributeTypeMap = [
    {
        name: "rowFormat",
        baseName: "RowFormat",
        type: "TableRowFormat",
    }
];
