// asposewordscloud/dist/model/tableRow.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const nodeLink_1 = require("./nodeLink");
const tableCell_1 = require("./tableCell");
const tableRowFormat_1 = require("./tableRowFormat");
exports.importsMapTableRow = {
    NodeLink: nodeLink_1.NodeLink,
    TableCell: tableCell_1.TableCell,
    TableRowFormat: tableRowFormat_1.TableRowFormat,
};
/**
 * Table row element.
 */
class TableRow extends nodeLink_1.NodeLink {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(TableRow.attributeTypeMap);
    }
}
exports.TableRow = TableRow;
/**
 * Attribute type map
 */
TableRow.attributeTypeMap = [
    {
        name: "rowFormat",
        baseName: "RowFormat",
        type: "TableRowFormat",
    },
    {
        name: "tableCellList",
        baseName: "TableCellList",
        type: "Array<TableCell>",
    }
];
