// asposewordscloud/dist/model/tableLinkCollection.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const linkElement_1 = require("./linkElement");
const tableLink_1 = require("./tableLink");
const wordsApiLink_1 = require("./wordsApiLink");
exports.importsMapTableLinkCollection = {
    LinkElement: linkElement_1.LinkElement,
    TableLink: tableLink_1.TableLink,
    WordsApiLink: wordsApiLink_1.WordsApiLink,
};
/**
 * Collection of links to tables.
 */
class TableLinkCollection extends linkElement_1.LinkElement {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(TableLinkCollection.attributeTypeMap);
    }
}
exports.TableLinkCollection = TableLinkCollection;
/**
 * Attribute type map
 */
TableLinkCollection.attributeTypeMap = [
    {
        name: "tableLinkList",
        baseName: "TableLinkList",
        type: "Array<TableLink>",
    }
];
