// asposewordscloud/dist/model/tableProperties.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const linkElement_1 = require("./linkElement");
const preferredWidth_1 = require("./preferredWidth");
const wordsApiLink_1 = require("./wordsApiLink");
exports.importsMapTableProperties = {
    LinkElement: linkElement_1.LinkElement,
    PreferredWidth: preferredWidth_1.PreferredWidth,
    WordsApiLink: wordsApiLink_1.WordsApiLink,
};
/**
 * Represents the table properties.
 */
class TableProperties extends linkElement_1.LinkElement {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(TableProperties.attributeTypeMap);
    }
}
exports.TableProperties = TableProperties;
/**
 * Attribute type map
 */
TableProperties.attributeTypeMap = [
    {
        name: "alignment",
        baseName: "Alignment",
        type: "TableProperties.AlignmentEnum",
    },
    {
        name: "allowAutoFit",
        baseName: "AllowAutoFit",
        type: "boolean",
    },
    {
        name: "bidi",
        baseName: "Bidi",
        type: "boolean",
    },
    {
        name: "bottomPadding",
        baseName: "BottomPadding",
        type: "number",
    },
    {
        name: "cellSpacing",
        baseName: "CellSpacing",
        type: "number",
    },
    {
        name: "leftIndent",
        baseName: "LeftIndent",
        type: "number",
    },
    {
        name: "leftPadding",
        baseName: "LeftPadding",
        type: "number",
    },
    {
        name: "preferredWidth",
        baseName: "PreferredWidth",
        type: "PreferredWidth",
    },
    {
        name: "rightPadding",
        baseName: "RightPadding",
        type: "number",
    },
    {
        name: "styleIdentifier",
        baseName: "StyleIdentifier",
        type: "TableProperties.StyleIdentifierEnum",
    },
    {
        name: "styleName",
        baseName: "StyleName",
        type: "string",
    },
    {
        name: "styleOptions",
        baseName: "StyleOptions",
        type: "TableProperties.StyleOptionsEnum",
    },
    {
        name: "textWrapping",
        baseName: "TextWrapping",
        type: "TableProperties.TextWrappingEnum",
    },
    {
        name: "topPadding",
        baseName: "TopPadding",
        type: "number",
    }
];
/**
 * Enums for TableProperties
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (TableProperties) {
    let AlignmentEnum;
    (function (AlignmentEnum) {
        AlignmentEnum[AlignmentEnum["Left"] = 'Left'] = "Left";
        AlignmentEnum[AlignmentEnum["Center"] = 'Center'] = "Center";
        AlignmentEnum[AlignmentEnum["Right"] = 'Right'] = "Right";
    })(AlignmentEnum = TableProperties.AlignmentEnum || (TableProperties.AlignmentEnum = {}));
    let StyleIdentifierEnum;
    (function (StyleIdentifierEnum) {
        StyleIdentifierEnum[StyleIdentifierEnum["Normal"] = 'Normal'] = "Normal";
        StyleIdentifierEnum[StyleIdentifierEnum["Heading1"] = 'Heading1'] = "Heading1";
        StyleIdentifierEnum[StyleIdentifierEnum["Heading2"] = 'Heading2'] = "Heading2";
        StyleIdentifierEnum[StyleIdentifierEnum["Heading3"] = 'Heading3'] = "Heading3";
        StyleIdentifierEnum[StyleIdentifierEnum["Heading4"] = 'Heading4'] = "Heading4";
        StyleIdentifierEnum[StyleIdentifierEnum["Heading5"] = 'Heading5'] = "Heading5";
        StyleIdentifierEnum[StyleIdentifierEnum["Heading6"] = 'Heading6'] = "Heading6";
        StyleIdentifierEnum[StyleIdentifierEnum["Heading7"] = 'Heading7'] = "Heading7";
        StyleIdentifierEnum[StyleIdentifierEnum["Heading8"] = 'Heading8'] = "Heading8";
        StyleIdentifierEnum[StyleIdentifierEnum["Heading9"] = 'Heading9'] = "Heading9";
        StyleIdentifierEnum[StyleIdentifierEnum["Index1"] = 'Index1'] = "Index1";
        StyleIdentifierEnum[StyleIdentifierEnum["Index2"] = 'Index2'] = "Index2";
        StyleIdentifierEnum[StyleIdentifierEnum["Index3"] = 'Index3'] = "Index3";
        StyleIdentifierEnum[StyleIdentifierEnum["Index4"] = 'Index4'] = "Index4";
        StyleIdentifierEnum[StyleIdentifierEnum["Index5"] = 'Index5'] = "Index5";
        StyleIdentifierEnum[StyleIdentifierEnum["Index6"] = 'Index6'] = "Index6";
        StyleIdentifierEnum[StyleIdentifierEnum["Index7"] = 'Index7'] = "Index7";
        StyleIdentifierEnum[StyleIdentifierEnum["Index8"] = 'Index8'] = "Index8";
        StyleIdentifierEnum[StyleIdentifierEnum["Index9"] = 'Index9'] = "Index9";
        StyleIdentifierEnum[StyleIdentifierEnum["Toc1"] = 'Toc1'] = "Toc1";
        StyleIdentifierEnum[StyleIdentifierEnum["Toc2"] = 'Toc2'] = "Toc2";
        StyleIdentifierEnum[StyleIdentifierEnum["Toc3"] = 'Toc3'] = "Toc3";
        StyleIdentifierEnum[StyleIdentifierEnum["Toc4"] = 'Toc4'] = "Toc4";
        StyleIdentifierEnum[StyleIdentifierEnum["Toc5"] = 'Toc5'] = "Toc5";
        StyleIdentifierEnum[StyleIdentifierEnum["Toc6"] = 'Toc6'] = "Toc6";
        StyleIdentifierEnum[StyleIdentifierEnum["Toc7"] = 'Toc7'] = "Toc7";
        StyleIdentifierEnum[StyleIdentifierEnum["Toc8"] = 'Toc8'] = "Toc8";
        StyleIdentifierEnum[StyleIdentifierEnum["Toc9"] = 'Toc9'] = "Toc9";
        StyleIdentifierEnum[StyleIdentifierEnum["NormalIndent"] = 'NormalIndent'] = "NormalIndent";
        StyleIdentifierEnum[StyleIdentifierEnum["FootnoteText"] = 'FootnoteText'] = "FootnoteText";
        StyleIdentifierEnum[StyleIdentifierEnum["CommentText"] = 'CommentText'] = "CommentText";
        StyleIdentifierEnum[StyleIdentifierEnum["Header"] = 'Header'] = "Header";
        StyleIdentifierEnum[StyleIdentifierEnum["Footer"] = 'Footer'] = "Footer";
        StyleIdentifierEnum[StyleIdentifierEnum["IndexHeading"] = 'IndexHeading'] = "IndexHeading";
        StyleIdentifierEnum[StyleIdentifierEnum["Caption"] = 'Caption'] = "Caption";
        StyleIdentifierEnum[StyleIdentifierEnum["TableOfFigures"] = 'TableOfFigures'] = "TableOfFigures";
        StyleIdentifierEnum[StyleIdentifierEnum["EnvelopeAddress"] = 'EnvelopeAddress'] = "EnvelopeAddress";
        StyleIdentifierEnum[StyleIdentifierEnum["EnvelopeReturn"] = 'EnvelopeReturn'] = "EnvelopeReturn";
        StyleIdentifierEnum[StyleIdentifierEnum["FootnoteReference"] = 'FootnoteReference'] = "FootnoteReference";
        StyleIdentifierEnum[StyleIdentifierEnum["CommentReference"] = 'CommentReference'] = "CommentReference";
        StyleIdentifierEnum[StyleIdentifierEnum["LineNumber"] = 'LineNumber'] = "LineNumber";
        StyleIdentifierEnum[StyleIdentifierEnum["PageNumber"] = 'PageNumber'] = "PageNumber";
        StyleIdentifierEnum[StyleIdentifierEnum["EndnoteReference"] = 'EndnoteReference'] = "EndnoteReference";
        StyleIdentifierEnum[StyleIdentifierEnum["EndnoteText"] = 'EndnoteText'] = "EndnoteText";
        StyleIdentifierEnum[StyleIdentifierEnum["TableOfAuthorities"] = 'TableOfAuthorities'] = "TableOfAuthorities";
        StyleIdentifierEnum[StyleIdentifierEnum["Macro"] = 'Macro'] = "Macro";
        StyleIdentifierEnum[StyleIdentifierEnum["ToaHeading"] = 'ToaHeading'] = "ToaHeading";
        StyleIdentifierEnum[StyleIdentifierEnum["List"] = 'List'] = "List";
        StyleIdentifierEnum[StyleIdentifierEnum["ListBullet"] = 'ListBullet'] = "ListBullet";
        StyleIdentifierEnum[StyleIdentifierEnum["ListNumber"] = 'ListNumber'] = "ListNumber";
        StyleIdentifierEnum[StyleIdentifierEnum["List2"] = 'List2'] = "List2";
        StyleIdentifierEnum[StyleIdentifierEnum["List3"] = 'List3'] = "List3";
        StyleIdentifierEnum[StyleIdentifierEnum["List4"] = 'List4'] = "List4";
        StyleIdentifierEnum[StyleIdentifierEnum["List5"] = 'List5'] = "List5";
        StyleIdentifierEnum[StyleIdentifierEnum["ListBullet2"] = 'ListBullet2'] = "ListBullet2";
        StyleIdentifierEnum[StyleIdentifierEnum["ListBullet3"] = 'ListBullet3'] = "ListBullet3";
        StyleIdentifierEnum[StyleIdentifierEnum["ListBullet4"] = 'ListBullet4'] = "ListBullet4";
        StyleIdentifierEnum[StyleIdentifierEnum["ListBullet5"] = 'ListBullet5'] = "ListBullet5";
        StyleIdentifierEnum[StyleIdentifierEnum["ListNumber2"] = 'ListNumber2'] = "ListNumber2";
        StyleIdentifierEnum[StyleIdentifierEnum["ListNumber3"] = 'ListNumber3'] = "ListNumber3";
        StyleIdentifierEnum[StyleIdentifierEnum["ListNumber4"] = 'ListNumber4'] = "ListNumber4";
        StyleIdentifierEnum[StyleIdentifierEnum["ListNumber5"] = 'ListNumber5'] = "ListNumber5";
        StyleIdentifierEnum[StyleIdentifierEnum["Title"] = 'Title'] = "Title";
        StyleIdentifierEnum[StyleIdentifierEnum["Closing"] = 'Closing'] = "Closing";
        StyleIdentifierEnum[StyleIdentifierEnum["Signature"] = 'Signature'] = "Signature";
        StyleIdentifierEnum[StyleIdentifierEnum["DefaultParagraphFont"] = 'DefaultParagraphFont'] = "DefaultParagraphFont";
        StyleIdentifierEnum[StyleIdentifierEnum["BodyText"] = 'BodyText'] = "BodyText";
        StyleIdentifierEnum[StyleIdentifierEnum["BodyTextInd"] = 'BodyTextInd'] = "BodyTextInd";
        StyleIdentifierEnum[StyleIdentifierEnum["ListContinue"] = 'ListContinue'] = "ListContinue";
        StyleIdentifierEnum[StyleIdentifierEnum["ListContinue2"] = 'ListContinue2'] = "ListContinue2";
        StyleIdentifierEnum[StyleIdentifierEnum["ListContinue3"] = 'ListContinue3'] = "ListContinue3";
        StyleIdentifierEnum[StyleIdentifierEnum["ListContinue4"] = 'ListContinue4'] = "ListContinue4";
        StyleIdentifierEnum[StyleIdentifierEnum["ListContinue5"] = 'ListContinue5'] = "ListContinue5";
        StyleIdentifierEnum[StyleIdentifierEnum["MessageHeader"] = 'MessageHeader'] = "MessageHeader";
        StyleIdentifierEnum[StyleIdentifierEnum["Subtitle"] = 'Subtitle'] = "Subtitle";
        StyleIdentifierEnum[StyleIdentifierEnum["Salutation"] = 'Salutation'] = "Salutation";
        StyleIdentifierEnum[StyleIdentifierEnum["Date"] = 'Date'] = "Date";
        StyleIdentifierEnum[StyleIdentifierEnum["BodyText1I"] = 'BodyText1I'] = "BodyText1I";
        StyleIdentifierEnum[StyleIdentifierEnum["BodyText1I2"] = 'BodyText1I2'] = "BodyText1I2";
        StyleIdentifierEnum[StyleIdentifierEnum["NoteHeading"] = 'NoteHeading'] = "NoteHeading";
        StyleIdentifierEnum[StyleIdentifierEnum["BodyText2"] = 'BodyText2'] = "BodyText2";
        StyleIdentifierEnum[StyleIdentifierEnum["BodyText3"] = 'BodyText3'] = "BodyText3";
        StyleIdentifierEnum[StyleIdentifierEnum["BodyTextInd2"] = 'BodyTextInd2'] = "BodyTextInd2";
        StyleIdentifierEnum[StyleIdentifierEnum["BodyTextInd3"] = 'BodyTextInd3'] = "BodyTextInd3";
        StyleIdentifierEnum[StyleIdentifierEnum["BlockText"] = 'BlockText'] = "BlockText";
        StyleIdentifierEnum[StyleIdentifierEnum["Hyperlink"] = 'Hyperlink'] = "Hyperlink";
        StyleIdentifierEnum[StyleIdentifierEnum["FollowedHyperlink"] = 'FollowedHyperlink'] = "FollowedHyperlink";
        StyleIdentifierEnum[StyleIdentifierEnum["Strong"] = 'Strong'] = "Strong";
        StyleIdentifierEnum[StyleIdentifierEnum["Emphasis"] = 'Emphasis'] = "Emphasis";
        StyleIdentifierEnum[StyleIdentifierEnum["DocumentMap"] = 'DocumentMap'] = "DocumentMap";
        StyleIdentifierEnum[StyleIdentifierEnum["PlainText"] = 'PlainText'] = "PlainText";
        StyleIdentifierEnum[StyleIdentifierEnum["EmailSignature"] = 'EmailSignature'] = "EmailSignature";
        StyleIdentifierEnum[StyleIdentifierEnum["HtmlTopOfForm"] = 'HtmlTopOfForm'] = "HtmlTopOfForm";
        StyleIdentifierEnum[StyleIdentifierEnum["HtmlBottomOfForm"] = 'HtmlBottomOfForm'] = "HtmlBottomOfForm";
        StyleIdentifierEnum[StyleIdentifierEnum["NormalWeb"] = 'NormalWeb'] = "NormalWeb";
        StyleIdentifierEnum[StyleIdentifierEnum["HtmlAcronym"] = 'HtmlAcronym'] = "HtmlAcronym";
        StyleIdentifierEnum[StyleIdentifierEnum["HtmlAddress"] = 'HtmlAddress'] = "HtmlAddress";
        StyleIdentifierEnum[StyleIdentifierEnum["HtmlCite"] = 'HtmlCite'] = "HtmlCite";
        StyleIdentifierEnum[StyleIdentifierEnum["HtmlCode"] = 'HtmlCode'] = "HtmlCode";
        StyleIdentifierEnum[StyleIdentifierEnum["HtmlDefinition"] = 'HtmlDefinition'] = "HtmlDefinition";
        StyleIdentifierEnum[StyleIdentifierEnum["HtmlKeyboard"] = 'HtmlKeyboard'] = "HtmlKeyboard";
        StyleIdentifierEnum[StyleIdentifierEnum["HtmlPreformatted"] = 'HtmlPreformatted'] = "HtmlPreformatted";
        StyleIdentifierEnum[StyleIdentifierEnum["HtmlSample"] = 'HtmlSample'] = "HtmlSample";
        StyleIdentifierEnum[StyleIdentifierEnum["HtmlTypewriter"] = 'HtmlTypewriter'] = "HtmlTypewriter";
        StyleIdentifierEnum[StyleIdentifierEnum["HtmlVariable"] = 'HtmlVariable'] = "HtmlVariable";
        StyleIdentifierEnum[StyleIdentifierEnum["TableNormal"] = 'TableNormal'] = "TableNormal";
        StyleIdentifierEnum[StyleIdentifierEnum["CommentSubject"] = 'CommentSubject'] = "CommentSubject";
        StyleIdentifierEnum[StyleIdentifierEnum["NoList"] = 'NoList'] = "NoList";
        StyleIdentifierEnum[StyleIdentifierEnum["OutlineList1"] = 'OutlineList1'] = "OutlineList1";
        StyleIdentifierEnum[StyleIdentifierEnum["OutlineList2"] = 'OutlineList2'] = "OutlineList2";
        StyleIdentifierEnum[StyleIdentifierEnum["OutlineList3"] = 'OutlineList3'] = "OutlineList3";
        StyleIdentifierEnum[StyleIdentifierEnum["TableSimple1"] = 'TableSimple1'] = "TableSimple1";
        StyleIdentifierEnum[StyleIdentifierEnum["TableSimple2"] = 'TableSimple2'] = "TableSimple2";
        StyleIdentifierEnum[StyleIdentifierEnum["TableSimple3"] = 'TableSimple3'] = "TableSimple3";
        StyleIdentifierEnum[StyleIdentifierEnum["TableClassic1"] = 'TableClassic1'] = "TableClassic1";
        StyleIdentifierEnum[StyleIdentifierEnum["TableClassic2"] = 'TableClassic2'] = "TableClassic2";
        StyleIdentifierEnum[StyleIdentifierEnum["TableClassic3"] = 'TableClassic3'] = "TableClassic3";
        StyleIdentifierEnum[StyleIdentifierEnum["TableClassic4"] = 'TableClassic4'] = "TableClassic4";
        StyleIdentifierEnum[StyleIdentifierEnum["TableColorful1"] = 'TableColorful1'] = "TableColorful1";
        StyleIdentifierEnum[StyleIdentifierEnum["TableColorful2"] = 'TableColorful2'] = "TableColorful2";
        StyleIdentifierEnum[StyleIdentifierEnum["TableColorful3"] = 'TableColorful3'] = "TableColorful3";
        StyleIdentifierEnum[StyleIdentifierEnum["TableColumns1"] = 'TableColumns1'] = "TableColumns1";
        StyleIdentifierEnum[StyleIdentifierEnum["TableColumns2"] = 'TableColumns2'] = "TableColumns2";
        StyleIdentifierEnum[StyleIdentifierEnum["TableColumns3"] = 'TableColumns3'] = "TableColumns3";
        StyleIdentifierEnum[StyleIdentifierEnum["TableColumns4"] = 'TableColumns4'] = "TableColumns4";
        StyleIdentifierEnum[StyleIdentifierEnum["TableColumns5"] = 'TableColumns5'] = "TableColumns5";
        StyleIdentifierEnum[StyleIdentifierEnum["TableGrid1"] = 'TableGrid1'] = "TableGrid1";
        StyleIdentifierEnum[StyleIdentifierEnum["TableGrid2"] = 'TableGrid2'] = "TableGrid2";
        StyleIdentifierEnum[StyleIdentifierEnum["TableGrid3"] = 'TableGrid3'] = "TableGrid3";
        StyleIdentifierEnum[StyleIdentifierEnum["TableGrid4"] = 'TableGrid4'] = "TableGrid4";
        StyleIdentifierEnum[StyleIdentifierEnum["TableGrid5"] = 'TableGrid5'] = "TableGrid5";
        StyleIdentifierEnum[StyleIdentifierEnum["TableGrid6"] = 'TableGrid6'] = "TableGrid6";
        StyleIdentifierEnum[StyleIdentifierEnum["TableGrid7"] = 'TableGrid7'] = "TableGrid7";
        StyleIdentifierEnum[StyleIdentifierEnum["TableGrid8"] = 'TableGrid8'] = "TableGrid8";
        StyleIdentifierEnum[StyleIdentifierEnum["TableList1"] = 'TableList1'] = "TableList1";
        StyleIdentifierEnum[StyleIdentifierEnum["TableList2"] = 'TableList2'] = "TableList2";
        StyleIdentifierEnum[StyleIdentifierEnum["TableList3"] = 'TableList3'] = "TableList3";
        StyleIdentifierEnum[StyleIdentifierEnum["TableList4"] = 'TableList4'] = "TableList4";
        StyleIdentifierEnum[StyleIdentifierEnum["TableList5"] = 'TableList5'] = "TableList5";
        StyleIdentifierEnum[StyleIdentifierEnum["TableList6"] = 'TableList6'] = "TableList6";
        StyleIdentifierEnum[StyleIdentifierEnum["TableList7"] = 'TableList7'] = "TableList7";
        StyleIdentifierEnum[StyleIdentifierEnum["TableList8"] = 'TableList8'] = "TableList8";
        StyleIdentifierEnum[StyleIdentifierEnum["Table3DEffects1"] = 'Table3DEffects1'] = "Table3DEffects1";
        StyleIdentifierEnum[StyleIdentifierEnum["Table3DEffects2"] = 'Table3DEffects2'] = "Table3DEffects2";
        StyleIdentifierEnum[StyleIdentifierEnum["Table3DEffects3"] = 'Table3DEffects3'] = "Table3DEffects3";
        StyleIdentifierEnum[StyleIdentifierEnum["TableContemporary"] = 'TableContemporary'] = "TableContemporary";
        StyleIdentifierEnum[StyleIdentifierEnum["TableElegant"] = 'TableElegant'] = "TableElegant";
        StyleIdentifierEnum[StyleIdentifierEnum["TableProfessional"] = 'TableProfessional'] = "TableProfessional";
        StyleIdentifierEnum[StyleIdentifierEnum["TableSubtle1"] = 'TableSubtle1'] = "TableSubtle1";
        StyleIdentifierEnum[StyleIdentifierEnum["TableSubtle2"] = 'TableSubtle2'] = "TableSubtle2";
        StyleIdentifierEnum[StyleIdentifierEnum["TableWeb1"] = 'TableWeb1'] = "TableWeb1";
        StyleIdentifierEnum[StyleIdentifierEnum["TableWeb2"] = 'TableWeb2'] = "TableWeb2";
        StyleIdentifierEnum[StyleIdentifierEnum["TableWeb3"] = 'TableWeb3'] = "TableWeb3";
        StyleIdentifierEnum[StyleIdentifierEnum["BalloonText"] = 'BalloonText'] = "BalloonText";
        StyleIdentifierEnum[StyleIdentifierEnum["TableGrid"] = 'TableGrid'] = "TableGrid";
        StyleIdentifierEnum[StyleIdentifierEnum["TableTheme"] = 'TableTheme'] = "TableTheme";
        StyleIdentifierEnum[StyleIdentifierEnum["PlaceholderText"] = 'PlaceholderText'] = "PlaceholderText";
        StyleIdentifierEnum[StyleIdentifierEnum["NoSpacing"] = 'NoSpacing'] = "NoSpacing";
        StyleIdentifierEnum[StyleIdentifierEnum["LightShading"] = 'LightShading'] = "LightShading";
        StyleIdentifierEnum[StyleIdentifierEnum["LightList"] = 'LightList'] = "LightList";
        StyleIdentifierEnum[StyleIdentifierEnum["LightGrid"] = 'LightGrid'] = "LightGrid";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumShading1"] = 'MediumShading1'] = "MediumShading1";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumShading2"] = 'MediumShading2'] = "MediumShading2";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumList1"] = 'MediumList1'] = "MediumList1";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumList2"] = 'MediumList2'] = "MediumList2";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid1"] = 'MediumGrid1'] = "MediumGrid1";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid2"] = 'MediumGrid2'] = "MediumGrid2";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid3"] = 'MediumGrid3'] = "MediumGrid3";
        StyleIdentifierEnum[StyleIdentifierEnum["DarkList"] = 'DarkList'] = "DarkList";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulShading"] = 'ColorfulShading'] = "ColorfulShading";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulList"] = 'ColorfulList'] = "ColorfulList";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulGrid"] = 'ColorfulGrid'] = "ColorfulGrid";
        StyleIdentifierEnum[StyleIdentifierEnum["LightShadingAccent1"] = 'LightShadingAccent1'] = "LightShadingAccent1";
        StyleIdentifierEnum[StyleIdentifierEnum["LightListAccent1"] = 'LightListAccent1'] = "LightListAccent1";
        StyleIdentifierEnum[StyleIdentifierEnum["LightGridAccent1"] = 'LightGridAccent1'] = "LightGridAccent1";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumShading1Accent1"] = 'MediumShading1Accent1'] = "MediumShading1Accent1";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumShading2Accent1"] = 'MediumShading2Accent1'] = "MediumShading2Accent1";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumList1Accent1"] = 'MediumList1Accent1'] = "MediumList1Accent1";
        StyleIdentifierEnum[StyleIdentifierEnum["Revision"] = 'Revision'] = "Revision";
        StyleIdentifierEnum[StyleIdentifierEnum["ListParagraph"] = 'ListParagraph'] = "ListParagraph";
        StyleIdentifierEnum[StyleIdentifierEnum["Quote"] = 'Quote'] = "Quote";
        StyleIdentifierEnum[StyleIdentifierEnum["IntenseQuote"] = 'IntenseQuote'] = "IntenseQuote";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumList2Accent1"] = 'MediumList2Accent1'] = "MediumList2Accent1";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid1Accent1"] = 'MediumGrid1Accent1'] = "MediumGrid1Accent1";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid2Accent1"] = 'MediumGrid2Accent1'] = "MediumGrid2Accent1";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid3Accent1"] = 'MediumGrid3Accent1'] = "MediumGrid3Accent1";
        StyleIdentifierEnum[StyleIdentifierEnum["DarkListAccent1"] = 'DarkListAccent1'] = "DarkListAccent1";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulShadingAccent1"] = 'ColorfulShadingAccent1'] = "ColorfulShadingAccent1";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulListAccent1"] = 'ColorfulListAccent1'] = "ColorfulListAccent1";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulGridAccent1"] = 'ColorfulGridAccent1'] = "ColorfulGridAccent1";
        StyleIdentifierEnum[StyleIdentifierEnum["LightShadingAccent2"] = 'LightShadingAccent2'] = "LightShadingAccent2";
        StyleIdentifierEnum[StyleIdentifierEnum["LightListAccent2"] = 'LightListAccent2'] = "LightListAccent2";
        StyleIdentifierEnum[StyleIdentifierEnum["LightGridAccent2"] = 'LightGridAccent2'] = "LightGridAccent2";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumShading1Accent2"] = 'MediumShading1Accent2'] = "MediumShading1Accent2";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumShading2Accent2"] = 'MediumShading2Accent2'] = "MediumShading2Accent2";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumList1Accent2"] = 'MediumList1Accent2'] = "MediumList1Accent2";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumList2Accent2"] = 'MediumList2Accent2'] = "MediumList2Accent2";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid1Accent2"] = 'MediumGrid1Accent2'] = "MediumGrid1Accent2";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid2Accent2"] = 'MediumGrid2Accent2'] = "MediumGrid2Accent2";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid3Accent2"] = 'MediumGrid3Accent2'] = "MediumGrid3Accent2";
        StyleIdentifierEnum[StyleIdentifierEnum["DarkListAccent2"] = 'DarkListAccent2'] = "DarkListAccent2";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulShadingAccent2"] = 'ColorfulShadingAccent2'] = "ColorfulShadingAccent2";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulListAccent2"] = 'ColorfulListAccent2'] = "ColorfulListAccent2";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulGridAccent2"] = 'ColorfulGridAccent2'] = "ColorfulGridAccent2";
        StyleIdentifierEnum[StyleIdentifierEnum["LightShadingAccent3"] = 'LightShadingAccent3'] = "LightShadingAccent3";
        StyleIdentifierEnum[StyleIdentifierEnum["LightListAccent3"] = 'LightListAccent3'] = "LightListAccent3";
        StyleIdentifierEnum[StyleIdentifierEnum["LightGridAccent3"] = 'LightGridAccent3'] = "LightGridAccent3";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumShading1Accent3"] = 'MediumShading1Accent3'] = "MediumShading1Accent3";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumShading2Accent3"] = 'MediumShading2Accent3'] = "MediumShading2Accent3";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumList1Accent3"] = 'MediumList1Accent3'] = "MediumList1Accent3";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumList2Accent3"] = 'MediumList2Accent3'] = "MediumList2Accent3";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid1Accent3"] = 'MediumGrid1Accent3'] = "MediumGrid1Accent3";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid2Accent3"] = 'MediumGrid2Accent3'] = "MediumGrid2Accent3";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid3Accent3"] = 'MediumGrid3Accent3'] = "MediumGrid3Accent3";
        StyleIdentifierEnum[StyleIdentifierEnum["DarkListAccent3"] = 'DarkListAccent3'] = "DarkListAccent3";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulShadingAccent3"] = 'ColorfulShadingAccent3'] = "ColorfulShadingAccent3";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulListAccent3"] = 'ColorfulListAccent3'] = "ColorfulListAccent3";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulGridAccent3"] = 'ColorfulGridAccent3'] = "ColorfulGridAccent3";
        StyleIdentifierEnum[StyleIdentifierEnum["LightShadingAccent4"] = 'LightShadingAccent4'] = "LightShadingAccent4";
        StyleIdentifierEnum[StyleIdentifierEnum["LightListAccent4"] = 'LightListAccent4'] = "LightListAccent4";
        StyleIdentifierEnum[StyleIdentifierEnum["LightGridAccent4"] = 'LightGridAccent4'] = "LightGridAccent4";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumShading1Accent4"] = 'MediumShading1Accent4'] = "MediumShading1Accent4";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumShading2Accent4"] = 'MediumShading2Accent4'] = "MediumShading2Accent4";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumList1Accent4"] = 'MediumList1Accent4'] = "MediumList1Accent4";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumList2Accent4"] = 'MediumList2Accent4'] = "MediumList2Accent4";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid1Accent4"] = 'MediumGrid1Accent4'] = "MediumGrid1Accent4";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid2Accent4"] = 'MediumGrid2Accent4'] = "MediumGrid2Accent4";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid3Accent4"] = 'MediumGrid3Accent4'] = "MediumGrid3Accent4";
        StyleIdentifierEnum[StyleIdentifierEnum["DarkListAccent4"] = 'DarkListAccent4'] = "DarkListAccent4";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulShadingAccent4"] = 'ColorfulShadingAccent4'] = "ColorfulShadingAccent4";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulListAccent4"] = 'ColorfulListAccent4'] = "ColorfulListAccent4";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulGridAccent4"] = 'ColorfulGridAccent4'] = "ColorfulGridAccent4";
        StyleIdentifierEnum[StyleIdentifierEnum["LightShadingAccent5"] = 'LightShadingAccent5'] = "LightShadingAccent5";
        StyleIdentifierEnum[StyleIdentifierEnum["LightListAccent5"] = 'LightListAccent5'] = "LightListAccent5";
        StyleIdentifierEnum[StyleIdentifierEnum["LightGridAccent5"] = 'LightGridAccent5'] = "LightGridAccent5";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumShading1Accent5"] = 'MediumShading1Accent5'] = "MediumShading1Accent5";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumShading2Accent5"] = 'MediumShading2Accent5'] = "MediumShading2Accent5";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumList1Accent5"] = 'MediumList1Accent5'] = "MediumList1Accent5";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumList2Accent5"] = 'MediumList2Accent5'] = "MediumList2Accent5";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid1Accent5"] = 'MediumGrid1Accent5'] = "MediumGrid1Accent5";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid2Accent5"] = 'MediumGrid2Accent5'] = "MediumGrid2Accent5";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid3Accent5"] = 'MediumGrid3Accent5'] = "MediumGrid3Accent5";
        StyleIdentifierEnum[StyleIdentifierEnum["DarkListAccent5"] = 'DarkListAccent5'] = "DarkListAccent5";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulShadingAccent5"] = 'ColorfulShadingAccent5'] = "ColorfulShadingAccent5";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulListAccent5"] = 'ColorfulListAccent5'] = "ColorfulListAccent5";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulGridAccent5"] = 'ColorfulGridAccent5'] = "ColorfulGridAccent5";
        StyleIdentifierEnum[StyleIdentifierEnum["LightShadingAccent6"] = 'LightShadingAccent6'] = "LightShadingAccent6";
        StyleIdentifierEnum[StyleIdentifierEnum["LightListAccent6"] = 'LightListAccent6'] = "LightListAccent6";
        StyleIdentifierEnum[StyleIdentifierEnum["LightGridAccent6"] = 'LightGridAccent6'] = "LightGridAccent6";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumShading1Accent6"] = 'MediumShading1Accent6'] = "MediumShading1Accent6";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumShading2Accent6"] = 'MediumShading2Accent6'] = "MediumShading2Accent6";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumList1Accent6"] = 'MediumList1Accent6'] = "MediumList1Accent6";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumList2Accent6"] = 'MediumList2Accent6'] = "MediumList2Accent6";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid1Accent6"] = 'MediumGrid1Accent6'] = "MediumGrid1Accent6";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid2Accent6"] = 'MediumGrid2Accent6'] = "MediumGrid2Accent6";
        StyleIdentifierEnum[StyleIdentifierEnum["MediumGrid3Accent6"] = 'MediumGrid3Accent6'] = "MediumGrid3Accent6";
        StyleIdentifierEnum[StyleIdentifierEnum["DarkListAccent6"] = 'DarkListAccent6'] = "DarkListAccent6";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulShadingAccent6"] = 'ColorfulShadingAccent6'] = "ColorfulShadingAccent6";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulListAccent6"] = 'ColorfulListAccent6'] = "ColorfulListAccent6";
        StyleIdentifierEnum[StyleIdentifierEnum["ColorfulGridAccent6"] = 'ColorfulGridAccent6'] = "ColorfulGridAccent6";
        StyleIdentifierEnum[StyleIdentifierEnum["SubtleEmphasis"] = 'SubtleEmphasis'] = "SubtleEmphasis";
        StyleIdentifierEnum[StyleIdentifierEnum["IntenseEmphasis"] = 'IntenseEmphasis'] = "IntenseEmphasis";
        StyleIdentifierEnum[StyleIdentifierEnum["SubtleReference"] = 'SubtleReference'] = "SubtleReference";
        StyleIdentifierEnum[StyleIdentifierEnum["IntenseReference"] = 'IntenseReference'] = "IntenseReference";
        StyleIdentifierEnum[StyleIdentifierEnum["BookTitle"] = 'BookTitle'] = "BookTitle";
        StyleIdentifierEnum[StyleIdentifierEnum["Bibliography"] = 'Bibliography'] = "Bibliography";
        StyleIdentifierEnum[StyleIdentifierEnum["TocHeading"] = 'TocHeading'] = "TocHeading";
        StyleIdentifierEnum[StyleIdentifierEnum["PlainTable1"] = 'PlainTable1'] = "PlainTable1";
        StyleIdentifierEnum[StyleIdentifierEnum["PlainTable2"] = 'PlainTable2'] = "PlainTable2";
        StyleIdentifierEnum[StyleIdentifierEnum["PlainTable3"] = 'PlainTable3'] = "PlainTable3";
        StyleIdentifierEnum[StyleIdentifierEnum["PlainTable4"] = 'PlainTable4'] = "PlainTable4";
        StyleIdentifierEnum[StyleIdentifierEnum["PlainTable5"] = 'PlainTable5'] = "PlainTable5";
        StyleIdentifierEnum[StyleIdentifierEnum["TableGridLight"] = 'TableGridLight'] = "TableGridLight";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable1Light"] = 'GridTable1Light'] = "GridTable1Light";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable2"] = 'GridTable2'] = "GridTable2";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable3"] = 'GridTable3'] = "GridTable3";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable4"] = 'GridTable4'] = "GridTable4";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable5Dark"] = 'GridTable5Dark'] = "GridTable5Dark";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable6Colorful"] = 'GridTable6Colorful'] = "GridTable6Colorful";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable7Colorful"] = 'GridTable7Colorful'] = "GridTable7Colorful";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable1LightAccent1"] = 'GridTable1LightAccent1'] = "GridTable1LightAccent1";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable2Accent1"] = 'GridTable2Accent1'] = "GridTable2Accent1";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable3Accent1"] = 'GridTable3Accent1'] = "GridTable3Accent1";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable4Accent1"] = 'GridTable4Accent1'] = "GridTable4Accent1";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable5DarkAccent1"] = 'GridTable5DarkAccent1'] = "GridTable5DarkAccent1";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable6ColorfulAccent1"] = 'GridTable6ColorfulAccent1'] = "GridTable6ColorfulAccent1";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable7ColorfulAccent1"] = 'GridTable7ColorfulAccent1'] = "GridTable7ColorfulAccent1";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable1LightAccent2"] = 'GridTable1LightAccent2'] = "GridTable1LightAccent2";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable2Accent2"] = 'GridTable2Accent2'] = "GridTable2Accent2";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable3Accent2"] = 'GridTable3Accent2'] = "GridTable3Accent2";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable4Accent2"] = 'GridTable4Accent2'] = "GridTable4Accent2";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable5DarkAccent2"] = 'GridTable5DarkAccent2'] = "GridTable5DarkAccent2";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable6ColorfulAccent2"] = 'GridTable6ColorfulAccent2'] = "GridTable6ColorfulAccent2";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable7ColorfulAccent2"] = 'GridTable7ColorfulAccent2'] = "GridTable7ColorfulAccent2";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable1LightAccent3"] = 'GridTable1LightAccent3'] = "GridTable1LightAccent3";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable2Accent3"] = 'GridTable2Accent3'] = "GridTable2Accent3";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable3Accent3"] = 'GridTable3Accent3'] = "GridTable3Accent3";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable4Accent3"] = 'GridTable4Accent3'] = "GridTable4Accent3";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable5DarkAccent3"] = 'GridTable5DarkAccent3'] = "GridTable5DarkAccent3";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable6ColorfulAccent3"] = 'GridTable6ColorfulAccent3'] = "GridTable6ColorfulAccent3";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable7ColorfulAccent3"] = 'GridTable7ColorfulAccent3'] = "GridTable7ColorfulAccent3";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable1LightAccent4"] = 'GridTable1LightAccent4'] = "GridTable1LightAccent4";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable2Accent4"] = 'GridTable2Accent4'] = "GridTable2Accent4";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable3Accent4"] = 'GridTable3Accent4'] = "GridTable3Accent4";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable4Accent4"] = 'GridTable4Accent4'] = "GridTable4Accent4";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable5DarkAccent4"] = 'GridTable5DarkAccent4'] = "GridTable5DarkAccent4";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable6ColorfulAccent4"] = 'GridTable6ColorfulAccent4'] = "GridTable6ColorfulAccent4";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable7ColorfulAccent4"] = 'GridTable7ColorfulAccent4'] = "GridTable7ColorfulAccent4";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable1LightAccent5"] = 'GridTable1LightAccent5'] = "GridTable1LightAccent5";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable2Accent5"] = 'GridTable2Accent5'] = "GridTable2Accent5";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable3Accent5"] = 'GridTable3Accent5'] = "GridTable3Accent5";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable4Accent5"] = 'GridTable4Accent5'] = "GridTable4Accent5";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable5DarkAccent5"] = 'GridTable5DarkAccent5'] = "GridTable5DarkAccent5";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable6ColorfulAccent5"] = 'GridTable6ColorfulAccent5'] = "GridTable6ColorfulAccent5";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable7ColorfulAccent5"] = 'GridTable7ColorfulAccent5'] = "GridTable7ColorfulAccent5";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable1LightAccent6"] = 'GridTable1LightAccent6'] = "GridTable1LightAccent6";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable2Accent6"] = 'GridTable2Accent6'] = "GridTable2Accent6";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable3Accent6"] = 'GridTable3Accent6'] = "GridTable3Accent6";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable4Accent6"] = 'GridTable4Accent6'] = "GridTable4Accent6";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable5DarkAccent6"] = 'GridTable5DarkAccent6'] = "GridTable5DarkAccent6";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable6ColorfulAccent6"] = 'GridTable6ColorfulAccent6'] = "GridTable6ColorfulAccent6";
        StyleIdentifierEnum[StyleIdentifierEnum["GridTable7ColorfulAccent6"] = 'GridTable7ColorfulAccent6'] = "GridTable7ColorfulAccent6";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable1Light"] = 'ListTable1Light'] = "ListTable1Light";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable2"] = 'ListTable2'] = "ListTable2";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable3"] = 'ListTable3'] = "ListTable3";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable4"] = 'ListTable4'] = "ListTable4";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable5Dark"] = 'ListTable5Dark'] = "ListTable5Dark";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable6Colorful"] = 'ListTable6Colorful'] = "ListTable6Colorful";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable7Colorful"] = 'ListTable7Colorful'] = "ListTable7Colorful";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable1LightAccent1"] = 'ListTable1LightAccent1'] = "ListTable1LightAccent1";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable2Accent1"] = 'ListTable2Accent1'] = "ListTable2Accent1";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable3Accent1"] = 'ListTable3Accent1'] = "ListTable3Accent1";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable4Accent1"] = 'ListTable4Accent1'] = "ListTable4Accent1";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable5DarkAccent1"] = 'ListTable5DarkAccent1'] = "ListTable5DarkAccent1";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable6ColorfulAccent1"] = 'ListTable6ColorfulAccent1'] = "ListTable6ColorfulAccent1";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable7ColorfulAccent1"] = 'ListTable7ColorfulAccent1'] = "ListTable7ColorfulAccent1";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable1LightAccent2"] = 'ListTable1LightAccent2'] = "ListTable1LightAccent2";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable2Accent2"] = 'ListTable2Accent2'] = "ListTable2Accent2";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable3Accent2"] = 'ListTable3Accent2'] = "ListTable3Accent2";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable4Accent2"] = 'ListTable4Accent2'] = "ListTable4Accent2";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable5DarkAccent2"] = 'ListTable5DarkAccent2'] = "ListTable5DarkAccent2";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable6ColorfulAccent2"] = 'ListTable6ColorfulAccent2'] = "ListTable6ColorfulAccent2";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable7ColorfulAccent2"] = 'ListTable7ColorfulAccent2'] = "ListTable7ColorfulAccent2";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable1LightAccent3"] = 'ListTable1LightAccent3'] = "ListTable1LightAccent3";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable2Accent3"] = 'ListTable2Accent3'] = "ListTable2Accent3";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable3Accent3"] = 'ListTable3Accent3'] = "ListTable3Accent3";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable4Accent3"] = 'ListTable4Accent3'] = "ListTable4Accent3";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable5DarkAccent3"] = 'ListTable5DarkAccent3'] = "ListTable5DarkAccent3";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable6ColorfulAccent3"] = 'ListTable6ColorfulAccent3'] = "ListTable6ColorfulAccent3";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable7ColorfulAccent3"] = 'ListTable7ColorfulAccent3'] = "ListTable7ColorfulAccent3";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable1LightAccent4"] = 'ListTable1LightAccent4'] = "ListTable1LightAccent4";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable2Accent4"] = 'ListTable2Accent4'] = "ListTable2Accent4";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable3Accent4"] = 'ListTable3Accent4'] = "ListTable3Accent4";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable4Accent4"] = 'ListTable4Accent4'] = "ListTable4Accent4";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable5DarkAccent4"] = 'ListTable5DarkAccent4'] = "ListTable5DarkAccent4";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable6ColorfulAccent4"] = 'ListTable6ColorfulAccent4'] = "ListTable6ColorfulAccent4";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable7ColorfulAccent4"] = 'ListTable7ColorfulAccent4'] = "ListTable7ColorfulAccent4";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable1LightAccent5"] = 'ListTable1LightAccent5'] = "ListTable1LightAccent5";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable2Accent5"] = 'ListTable2Accent5'] = "ListTable2Accent5";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable3Accent5"] = 'ListTable3Accent5'] = "ListTable3Accent5";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable4Accent5"] = 'ListTable4Accent5'] = "ListTable4Accent5";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable5DarkAccent5"] = 'ListTable5DarkAccent5'] = "ListTable5DarkAccent5";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable6ColorfulAccent5"] = 'ListTable6ColorfulAccent5'] = "ListTable6ColorfulAccent5";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable7ColorfulAccent5"] = 'ListTable7ColorfulAccent5'] = "ListTable7ColorfulAccent5";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable1LightAccent6"] = 'ListTable1LightAccent6'] = "ListTable1LightAccent6";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable2Accent6"] = 'ListTable2Accent6'] = "ListTable2Accent6";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable3Accent6"] = 'ListTable3Accent6'] = "ListTable3Accent6";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable4Accent6"] = 'ListTable4Accent6'] = "ListTable4Accent6";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable5DarkAccent6"] = 'ListTable5DarkAccent6'] = "ListTable5DarkAccent6";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable6ColorfulAccent6"] = 'ListTable6ColorfulAccent6'] = "ListTable6ColorfulAccent6";
        StyleIdentifierEnum[StyleIdentifierEnum["ListTable7ColorfulAccent6"] = 'ListTable7ColorfulAccent6'] = "ListTable7ColorfulAccent6";
        StyleIdentifierEnum[StyleIdentifierEnum["User"] = 'User'] = "User";
        StyleIdentifierEnum[StyleIdentifierEnum["Nil"] = 'Nil'] = "Nil";
    })(StyleIdentifierEnum = TableProperties.StyleIdentifierEnum || (TableProperties.StyleIdentifierEnum = {}));
    let StyleOptionsEnum;
    (function (StyleOptionsEnum) {
        StyleOptionsEnum[StyleOptionsEnum["None"] = 'None'] = "None";
        StyleOptionsEnum[StyleOptionsEnum["FirstRow"] = 'FirstRow'] = "FirstRow";
        StyleOptionsEnum[StyleOptionsEnum["LastRow"] = 'LastRow'] = "LastRow";
        StyleOptionsEnum[StyleOptionsEnum["FirstColumn"] = 'FirstColumn'] = "FirstColumn";
        StyleOptionsEnum[StyleOptionsEnum["LastColumn"] = 'LastColumn'] = "LastColumn";
        StyleOptionsEnum[StyleOptionsEnum["RowBands"] = 'RowBands'] = "RowBands";
        StyleOptionsEnum[StyleOptionsEnum["Default"] = 'Default'] = "Default";
        StyleOptionsEnum[StyleOptionsEnum["ColumnBands"] = 'ColumnBands'] = "ColumnBands";
        StyleOptionsEnum[StyleOptionsEnum["Default2003"] = 'Default2003'] = "Default2003";
    })(StyleOptionsEnum = TableProperties.StyleOptionsEnum || (TableProperties.StyleOptionsEnum = {}));
    let TextWrappingEnum;
    (function (TextWrappingEnum) {
        TextWrappingEnum[TextWrappingEnum["Default"] = 'Default'] = "Default";
        TextWrappingEnum[TextWrappingEnum["None"] = 'None'] = "None";
        TextWrappingEnum[TextWrappingEnum["Around"] = 'Around'] = "Around";
    })(TextWrappingEnum = TableProperties.TextWrappingEnum || (TableProperties.TextWrappingEnum = {}));
})(TableProperties = exports.TableProperties || (exports.TableProperties = {}));
// tslint:enable:quotemark
