// asposewordscloud/dist/model/tablePropertiesResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tableProperties_1 = require("./tableProperties");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapTablePropertiesResponse = {
    TableProperties: tableProperties_1.TableProperties,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/tables/{0}/properties.
 */
class TablePropertiesResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(TablePropertiesResponse.attributeTypeMap);
    }
}
exports.TablePropertiesResponse = TablePropertiesResponse;
/**
 * Attribute type map
 */
TablePropertiesResponse.attributeTypeMap = [
    {
        name: "properties",
        baseName: "Properties",
        type: "TableProperties",
    }
];
