// asposewordscloud/dist/model/tableLinkCollectionResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tableLinkCollection_1 = require("./tableLinkCollection");
const wordsResponse_1 = require("./wordsResponse");
exports.importsMapTableLinkCollectionResponse = {
    TableLinkCollection: tableLinkCollection_1.TableLinkCollection,
    WordsResponse: wordsResponse_1.WordsResponse,
};
/**
 * This response should be returned by the service when handling: GET https://api.aspose.cloud/v4.0/words/Test.doc/tables.
 */
class TableLinkCollectionResponse extends wordsResponse_1.WordsResponse {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(TableLinkCollectionResponse.attributeTypeMap);
    }
}
exports.TableLinkCollectionResponse = TableLinkCollectionResponse;
/**
 * Attribute type map
 */
TableLinkCollectionResponse.attributeTypeMap = [
    {
        name: "tables",
        baseName: "Tables",
        type: "TableLinkCollection",
    }
];
