// asposewordscloud/dist/model/tableCellFormat.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const linkElement_1 = require("./linkElement");
const preferredWidth_1 = require("./preferredWidth");
const wordsApiLink_1 = require("./wordsApiLink");
exports.importsMapTableCellFormat = {
    LinkElement: linkElement_1.LinkElement,
    PreferredWidth: preferredWidth_1.PreferredWidth,
    WordsApiLink: wordsApiLink_1.WordsApiLink,
};
/**
 * Represents all formatting for a table row.
 */
class TableCellFormat extends linkElement_1.LinkElement {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(TableCellFormat.attributeTypeMap);
    }
}
exports.TableCellFormat = TableCellFormat;
/**
 * Attribute type map
 */
TableCellFormat.attributeTypeMap = [
    {
        name: "bottomPadding",
        baseName: "BottomPadding",
        type: "number",
    },
    {
        name: "fitText",
        baseName: "FitText",
        type: "boolean",
    },
    {
        name: "horizontalMerge",
        baseName: "HorizontalMerge",
        type: "TableCellFormat.HorizontalMergeEnum",
    },
    {
        name: "leftPadding",
        baseName: "LeftPadding",
        type: "number",
    },
    {
        name: "orientation",
        baseName: "Orientation",
        type: "TableCellFormat.OrientationEnum",
    },
    {
        name: "preferredWidth",
        baseName: "PreferredWidth",
        type: "PreferredWidth",
    },
    {
        name: "rightPadding",
        baseName: "RightPadding",
        type: "number",
    },
    {
        name: "topPadding",
        baseName: "TopPadding",
        type: "number",
    },
    {
        name: "verticalAlignment",
        baseName: "VerticalAlignment",
        type: "TableCellFormat.VerticalAlignmentEnum",
    },
    {
        name: "verticalMerge",
        baseName: "VerticalMerge",
        type: "TableCellFormat.VerticalMergeEnum",
    },
    {
        name: "width",
        baseName: "Width",
        type: "number",
    },
    {
        name: "wrapText",
        baseName: "WrapText",
        type: "boolean",
    }
];
/**
 * Enums for TableCellFormat
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (TableCellFormat) {
    let HorizontalMergeEnum;
    (function (HorizontalMergeEnum) {
        HorizontalMergeEnum[HorizontalMergeEnum["None"] = 'None'] = "None";
        HorizontalMergeEnum[HorizontalMergeEnum["First"] = 'First'] = "First";
        HorizontalMergeEnum[HorizontalMergeEnum["Previous"] = 'Previous'] = "Previous";
    })(HorizontalMergeEnum = TableCellFormat.HorizontalMergeEnum || (TableCellFormat.HorizontalMergeEnum = {}));
    let OrientationEnum;
    (function (OrientationEnum) {
        OrientationEnum[OrientationEnum["Horizontal"] = 'Horizontal'] = "Horizontal";
        OrientationEnum[OrientationEnum["Downward"] = 'Downward'] = "Downward";
        OrientationEnum[OrientationEnum["Upward"] = 'Upward'] = "Upward";
        OrientationEnum[OrientationEnum["HorizontalRotatedFarEast"] = 'HorizontalRotatedFarEast'] = "HorizontalRotatedFarEast";
        OrientationEnum[OrientationEnum["VerticalFarEast"] = 'VerticalFarEast'] = "VerticalFarEast";
        OrientationEnum[OrientationEnum["VerticalRotatedFarEast"] = 'VerticalRotatedFarEast'] = "VerticalRotatedFarEast";
    })(OrientationEnum = TableCellFormat.OrientationEnum || (TableCellFormat.OrientationEnum = {}));
    let VerticalAlignmentEnum;
    (function (VerticalAlignmentEnum) {
        VerticalAlignmentEnum[VerticalAlignmentEnum["Top"] = 'Top'] = "Top";
        VerticalAlignmentEnum[VerticalAlignmentEnum["Center"] = 'Center'] = "Center";
        VerticalAlignmentEnum[VerticalAlignmentEnum["Bottom"] = 'Bottom'] = "Bottom";
    })(VerticalAlignmentEnum = TableCellFormat.VerticalAlignmentEnum || (TableCellFormat.VerticalAlignmentEnum = {}));
    let VerticalMergeEnum;
    (function (VerticalMergeEnum) {
        VerticalMergeEnum[VerticalMergeEnum["None"] = 'None'] = "None";
        VerticalMergeEnum[VerticalMergeEnum["First"] = 'First'] = "First";
        VerticalMergeEnum[VerticalMergeEnum["Previous"] = 'Previous'] = "Previous";
    })(VerticalMergeEnum = TableCellFormat.VerticalMergeEnum || (TableCellFormat.VerticalMergeEnum = {}));
})(TableCellFormat = exports.TableCellFormat || (exports.TableCellFormat = {}));
// tslint:enable:quotemark
