// asposewordscloud/dist/model/tabStopBase.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.importsMapTabStopBase = {};
/**
 * Base class for paragraph format tab stop DTO.
 */
class TabStopBase {
    constructor(init) {
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return TabStopBase.attributeTypeMap;
    }
}
exports.TabStopBase = TabStopBase;
/**
 * Attribute type map
 */
TabStopBase.attributeTypeMap = [
    {
        name: "alignment",
        baseName: "Alignment",
        type: "TabStopBase.AlignmentEnum",
    },
    {
        name: "leader",
        baseName: "Leader",
        type: "TabStopBase.LeaderEnum",
    },
    {
        name: "position",
        baseName: "Position",
        type: "number",
    }
];
/**
 * Enums for TabStopBase
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (TabStopBase) {
    let AlignmentEnum;
    (function (AlignmentEnum) {
        AlignmentEnum[AlignmentEnum["Left"] = 'Left'] = "Left";
        AlignmentEnum[AlignmentEnum["Center"] = 'Center'] = "Center";
        AlignmentEnum[AlignmentEnum["Right"] = 'Right'] = "Right";
        AlignmentEnum[AlignmentEnum["Decimal"] = 'Decimal'] = "Decimal";
        AlignmentEnum[AlignmentEnum["Bar"] = 'Bar'] = "Bar";
        AlignmentEnum[AlignmentEnum["List"] = 'List'] = "List";
        AlignmentEnum[AlignmentEnum["Clear"] = 'Clear'] = "Clear";
    })(AlignmentEnum = TabStopBase.AlignmentEnum || (TabStopBase.AlignmentEnum = {}));
    let LeaderEnum;
    (function (LeaderEnum) {
        LeaderEnum[LeaderEnum["None"] = 'None'] = "None";
        LeaderEnum[LeaderEnum["Dots"] = 'Dots'] = "Dots";
        LeaderEnum[LeaderEnum["Dashes"] = 'Dashes'] = "Dashes";
        LeaderEnum[LeaderEnum["Line"] = 'Line'] = "Line";
        LeaderEnum[LeaderEnum["Heavy"] = 'Heavy'] = "Heavy";
        LeaderEnum[LeaderEnum["MiddleDot"] = 'MiddleDot'] = "MiddleDot";
    })(LeaderEnum = TabStopBase.LeaderEnum || (TabStopBase.LeaderEnum = {}));
})(TabStopBase = exports.TabStopBase || (exports.TabStopBase = {}));
// tslint:enable:quotemark
