// asposewordscloud/dist/model/tiffSaveOptionsData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const graphicsQualityOptionsData_1 = require("./graphicsQualityOptionsData");
const imageSaveOptionsData_1 = require("./imageSaveOptionsData");
exports.importsMapTiffSaveOptionsData = {
    GraphicsQualityOptionsData: graphicsQualityOptionsData_1.GraphicsQualityOptionsData,
    ImageSaveOptionsData: imageSaveOptionsData_1.ImageSaveOptionsData,
};
/**
 * Container class for tiff save options.
 */
class TiffSaveOptionsData extends imageSaveOptionsData_1.ImageSaveOptionsData {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(TiffSaveOptionsData.attributeTypeMap);
    }
}
exports.TiffSaveOptionsData = TiffSaveOptionsData;
/**
 * Attribute type map
 */
TiffSaveOptionsData.attributeTypeMap = [
    {
        name: "thresholdForFloydSteinbergDithering",
        baseName: "ThresholdForFloydSteinbergDithering",
        type: "number",
    },
    {
        name: "tiffBinarizationMethod",
        baseName: "TiffBinarizationMethod",
        type: "string",
    },
    {
        name: "tiffCompression",
        baseName: "TiffCompression",
        type: "string",
    }
];
