// asposewordscloud/dist/model/xamlFlowSaveOptionsData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const saveOptionsData_1 = require("./saveOptionsData");
exports.importsMapXamlFlowSaveOptionsData = {
    SaveOptionsData: saveOptionsData_1.SaveOptionsData,
};
/**
 * container class for xaml flow save options.
 */
class XamlFlowSaveOptionsData extends saveOptionsData_1.SaveOptionsData {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(XamlFlowSaveOptionsData.attributeTypeMap);
    }
}
exports.XamlFlowSaveOptionsData = XamlFlowSaveOptionsData;
/**
 * Attribute type map
 */
XamlFlowSaveOptionsData.attributeTypeMap = [
    {
        name: "imagesFolder",
        baseName: "ImagesFolder",
        type: "string",
    },
    {
        name: "imagesFolderAlias",
        baseName: "ImagesFolderAlias",
        type: "string",
    }
];
// tslint:enable:quotemark
