// asposewordscloud/dist/model/xpsSaveOptionsData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const fixedPageSaveOptionsData_1 = require("./fixedPageSaveOptionsData");
const metafileRenderingOptionsData_1 = require("./metafileRenderingOptionsData");
const outlineOptionsData_1 = require("./outlineOptionsData");
exports.importsMapXpsSaveOptionsData = {
    FixedPageSaveOptionsData: fixedPageSaveOptionsData_1.FixedPageSaveOptionsData,
    MetafileRenderingOptionsData: metafileRenderingOptionsData_1.MetafileRenderingOptionsData,
    OutlineOptionsData: outlineOptionsData_1.OutlineOptionsData,
};
/**
 * Container class for xps save options.
 */
class XpsSaveOptionsData extends fixedPageSaveOptionsData_1.FixedPageSaveOptionsData {
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(XpsSaveOptionsData.attributeTypeMap);
    }
}
exports.XpsSaveOptionsData = XpsSaveOptionsData;
/**
 * Attribute type map
 */
XpsSaveOptionsData.attributeTypeMap = [
    {
        name: "bookmarksOutlineLevel",
        baseName: "BookmarksOutlineLevel",
        type: "number",
    },
    {
        name: "headingsOutlineLevels",
        baseName: "HeadingsOutlineLevels",
        type: "number",
    },
    {
        name: "outlineOptions",
        baseName: "OutlineOptions",
        type: "OutlineOptionsData",
    },
    {
        name: "useBookFoldPrintingSettings",
        baseName: "UseBookFoldPrintingSettings",
        type: "boolean",
    }
];
